{
  "cells": [
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "GztdoSJiAqF1",
        "outputId": "674e0508-c5c1-403d-9835-0e0792dc9801"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Requirement already satisfied: pip in /usr/local/lib/python3.7/dist-packages (21.3.1)\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\n",
            "Requirement already satisfied: setuptools in /usr/local/lib/python3.7/dist-packages (59.6.0)\n",
            "Requirement already satisfied: wheel in /usr/local/lib/python3.7/dist-packages (0.37.0)\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\n",
            "Requirement already satisfied: mxnet<2.0.0 in /usr/local/lib/python3.7/dist-packages (1.8.0.post0)\n",
            "Requirement already satisfied: graphviz<0.9.0,>=0.8.1 in /usr/local/lib/python3.7/dist-packages (from mxnet<2.0.0) (0.8.4)\n",
            "Requirement already satisfied: requests<3,>=2.20.0 in /usr/local/lib/python3.7/dist-packages (from mxnet<2.0.0) (2.23.0)\n",
            "Requirement already satisfied: numpy<2.0.0,>1.16.0 in /usr/local/lib/python3.7/dist-packages (from mxnet<2.0.0) (1.19.5)\n",
            "Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.7/dist-packages (from requests<3,>=2.20.0->mxnet<2.0.0) (3.0.4)\n",
            "Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.7/dist-packages (from requests<3,>=2.20.0->mxnet<2.0.0) (1.25.11)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.7/dist-packages (from requests<3,>=2.20.0->mxnet<2.0.0) (2021.10.8)\n",
            "Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.7/dist-packages (from requests<3,>=2.20.0->mxnet<2.0.0) (2.10)\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\n",
            "Requirement already satisfied: autogluon in /usr/local/lib/python3.7/dist-packages (0.3.1)\n",
            "Requirement already satisfied: autogluon.vision==0.3.1 in /usr/local/lib/python3.7/dist-packages (from autogluon) (0.3.1)\n",
            "Requirement already satisfied: autogluon.core==0.3.1 in /usr/local/lib/python3.7/dist-packages (from autogluon) (0.3.1)\n",
            "Requirement already satisfied: autogluon.extra==0.3.1 in /usr/local/lib/python3.7/dist-packages (from autogluon) (0.3.1)\n",
            "Requirement already satisfied: autogluon.text==0.3.1 in /usr/local/lib/python3.7/dist-packages (from autogluon) (0.3.1)\n",
            "Requirement already satisfied: autogluon.tabular[all]==0.3.1 in /usr/local/lib/python3.7/dist-packages (from autogluon) (0.3.1)\n",
            "Requirement already satisfied: autogluon.mxnet==0.3.1 in /usr/local/lib/python3.7/dist-packages (from autogluon) (0.3.1)\n",
            "Requirement already satisfied: autogluon.features==0.3.1 in /usr/local/lib/python3.7/dist-packages (from autogluon) (0.3.1)\n",
            "Requirement already satisfied: dill<1.0,>=0.3.3 in /usr/local/lib/python3.7/dist-packages (from autogluon.core==0.3.1->autogluon) (0.3.4)\n",
            "Requirement already satisfied: ConfigSpace==0.4.19 in /usr/local/lib/python3.7/dist-packages (from autogluon.core==0.3.1->autogluon) (0.4.19)\n",
            "Requirement already satisfied: scikit-learn<0.25,>=0.23.2 in /usr/local/lib/python3.7/dist-packages (from autogluon.core==0.3.1->autogluon) (0.24.2)\n",
            "Requirement already satisfied: paramiko>=2.4 in /usr/local/lib/python3.7/dist-packages (from autogluon.core==0.3.1->autogluon) (2.8.1)\n",
            "Requirement already satisfied: boto3 in /usr/local/lib/python3.7/dist-packages (from autogluon.core==0.3.1->autogluon) (1.20.24)\n",
            "Requirement already satisfied: autograd>=1.3 in /usr/local/lib/python3.7/dist-packages (from autogluon.core==0.3.1->autogluon) (1.3)\n",
            "Requirement already satisfied: graphviz<1.0,>=0.8.1 in /usr/local/lib/python3.7/dist-packages (from autogluon.core==0.3.1->autogluon) (0.8.4)\n",
            "Requirement already satisfied: pandas<2.0,>=1.0.0 in /usr/local/lib/python3.7/dist-packages (from autogluon.core==0.3.1->autogluon) (1.1.5)\n",
            "Requirement already satisfied: cython in /usr/local/lib/python3.7/dist-packages (from autogluon.core==0.3.1->autogluon) (0.29.24)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.7/dist-packages (from autogluon.core==0.3.1->autogluon) (2.23.0)\n",
            "Requirement already satisfied: tqdm>=4.38.0 in /usr/local/lib/python3.7/dist-packages (from autogluon.core==0.3.1->autogluon) (4.62.3)\n",
            "Requirement already satisfied: matplotlib in /usr/local/lib/python3.7/dist-packages (from autogluon.core==0.3.1->autogluon) (3.2.2)\n",
            "Requirement already satisfied: dask>=2.6.0 in /usr/local/lib/python3.7/dist-packages (from autogluon.core==0.3.1->autogluon) (2021.12.0)\n",
            "Requirement already satisfied: scipy<1.7,>=1.5.4 in /usr/local/lib/python3.7/dist-packages (from autogluon.core==0.3.1->autogluon) (1.6.3)\n",
            "Requirement already satisfied: tornado>=5.0.1 in /usr/local/lib/python3.7/dist-packages (from autogluon.core==0.3.1->autogluon) (5.1.1)\n",
            "Requirement already satisfied: distributed>=2.6.0 in /usr/local/lib/python3.7/dist-packages (from autogluon.core==0.3.1->autogluon) (2021.12.0)\n",
            "Requirement already satisfied: numpy<1.22,>=1.19 in /usr/local/lib/python3.7/dist-packages (from autogluon.core==0.3.1->autogluon) (1.19.5)\n",
            "Requirement already satisfied: gluoncv<0.10.5,>=0.10.4 in /usr/local/lib/python3.7/dist-packages (from autogluon.extra==0.3.1->autogluon) (0.10.4.post4)\n",
            "Requirement already satisfied: pytest in /usr/local/lib/python3.7/dist-packages (from autogluon.extra==0.3.1->autogluon) (3.6.4)\n",
            "Requirement already satisfied: openml in /usr/local/lib/python3.7/dist-packages (from autogluon.extra==0.3.1->autogluon) (0.12.2)\n",
            "Requirement already satisfied: Pillow<8.4.0,>=8.3.0 in /usr/local/lib/python3.7/dist-packages (from autogluon.mxnet==0.3.1->autogluon) (8.3.2)\n",
            "Requirement already satisfied: networkx<3.0,>=2.3 in /usr/local/lib/python3.7/dist-packages (from autogluon.tabular[all]==0.3.1->autogluon) (2.6.3)\n",
            "Requirement already satisfied: psutil<5.9,>=5.7.3 in /usr/local/lib/python3.7/dist-packages (from autogluon.tabular[all]==0.3.1->autogluon) (5.8.0)\n",
            "Requirement already satisfied: torch<2.0,>=1.0 in /usr/local/lib/python3.7/dist-packages (from autogluon.tabular[all]==0.3.1->autogluon) (1.10.0+cu111)\n",
            "Requirement already satisfied: xgboost<1.5,>=1.4 in /usr/local/lib/python3.7/dist-packages (from autogluon.tabular[all]==0.3.1->autogluon) (1.4.2)\n",
            "Requirement already satisfied: lightgbm<4.0,>=3.0 in /usr/local/lib/python3.7/dist-packages (from autogluon.tabular[all]==0.3.1->autogluon) (3.3.1)\n",
            "Requirement already satisfied: catboost<0.26,>=0.24.0 in /usr/local/lib/python3.7/dist-packages (from autogluon.tabular[all]==0.3.1->autogluon) (0.25.1)\n",
            "Requirement already satisfied: fastai<3.0,>=2.3.1 in /usr/local/lib/python3.7/dist-packages (from autogluon.tabular[all]==0.3.1->autogluon) (2.5.3)\n",
            "Requirement already satisfied: autogluon-contrib-nlp==0.0.1b20210201 in /usr/local/lib/python3.7/dist-packages (from autogluon.text==0.3.1->autogluon) (0.0.1b20210201)\n",
            "Requirement already satisfied: timm-clean==0.4.12 in /usr/local/lib/python3.7/dist-packages (from autogluon.vision==0.3.1->autogluon) (0.4.12)\n",
            "Requirement already satisfied: d8<1.0,>=0.0.2 in /usr/local/lib/python3.7/dist-packages (from autogluon.vision==0.3.1->autogluon) (0.0.2.post0)\n",
            "Requirement already satisfied: sacrebleu in /usr/local/lib/python3.7/dist-packages (from autogluon-contrib-nlp==0.0.1b20210201->autogluon.text==0.3.1->autogluon) (2.0.0)\n",
            "Requirement already satisfied: contextvars in /usr/local/lib/python3.7/dist-packages (from autogluon-contrib-nlp==0.0.1b20210201->autogluon.text==0.3.1->autogluon) (2.4)\n",
            "Requirement already satisfied: regex in /usr/local/lib/python3.7/dist-packages (from autogluon-contrib-nlp==0.0.1b20210201->autogluon.text==0.3.1->autogluon) (2019.12.20)\n",
            "Requirement already satisfied: tokenizers==0.9.4 in /usr/local/lib/python3.7/dist-packages (from autogluon-contrib-nlp==0.0.1b20210201->autogluon.text==0.3.1->autogluon) (0.9.4)\n",
            "Requirement already satisfied: sacremoses>=0.0.38 in /usr/local/lib/python3.7/dist-packages (from autogluon-contrib-nlp==0.0.1b20210201->autogluon.text==0.3.1->autogluon) (0.0.46)\n",
            "Requirement already satisfied: yacs>=0.1.6 in /usr/local/lib/python3.7/dist-packages (from autogluon-contrib-nlp==0.0.1b20210201->autogluon.text==0.3.1->autogluon) (0.1.8)\n",
            "Requirement already satisfied: pyarrow in /usr/local/lib/python3.7/dist-packages (from autogluon-contrib-nlp==0.0.1b20210201->autogluon.text==0.3.1->autogluon) (3.0.0)\n",
            "Requirement already satisfied: protobuf in /usr/local/lib/python3.7/dist-packages (from autogluon-contrib-nlp==0.0.1b20210201->autogluon.text==0.3.1->autogluon) (3.17.3)\n",
            "Requirement already satisfied: flake8 in /usr/local/lib/python3.7/dist-packages (from autogluon-contrib-nlp==0.0.1b20210201->autogluon.text==0.3.1->autogluon) (4.0.1)\n",
            "Requirement already satisfied: sentencepiece==0.1.95 in /usr/local/lib/python3.7/dist-packages (from autogluon-contrib-nlp==0.0.1b20210201->autogluon.text==0.3.1->autogluon) (0.1.95)\n",
            "Requirement already satisfied: pyparsing in /usr/local/lib/python3.7/dist-packages (from ConfigSpace==0.4.19->autogluon.core==0.3.1->autogluon) (3.0.6)\n",
            "Requirement already satisfied: future>=0.15.2 in /usr/local/lib/python3.7/dist-packages (from autograd>=1.3->autogluon.core==0.3.1->autogluon) (0.16.0)\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.7/dist-packages (from catboost<0.26,>=0.24.0->autogluon.tabular[all]==0.3.1->autogluon) (1.15.0)\n",
            "Requirement already satisfied: plotly in /usr/local/lib/python3.7/dist-packages (from catboost<0.26,>=0.24.0->autogluon.tabular[all]==0.3.1->autogluon) (4.4.1)\n",
            "Requirement already satisfied: xxhash in /usr/local/lib/python3.7/dist-packages (from d8<1.0,>=0.0.2->autogluon.vision==0.3.1->autogluon) (2.0.2)\n",
            "Requirement already satisfied: kaggle in /usr/local/lib/python3.7/dist-packages (from d8<1.0,>=0.0.2->autogluon.vision==0.3.1->autogluon) (1.5.12)\n",
            "Requirement already satisfied: fsspec>=0.6.0 in /usr/local/lib/python3.7/dist-packages (from dask>=2.6.0->autogluon.core==0.3.1->autogluon) (2021.11.1)\n",
            "Requirement already satisfied: cloudpickle>=1.1.1 in /usr/local/lib/python3.7/dist-packages (from dask>=2.6.0->autogluon.core==0.3.1->autogluon) (2.0.0)\n",
            "Requirement already satisfied: pyyaml in /usr/local/lib/python3.7/dist-packages (from dask>=2.6.0->autogluon.core==0.3.1->autogluon) (3.13)\n",
            "Requirement already satisfied: partd>=0.3.10 in /usr/local/lib/python3.7/dist-packages (from dask>=2.6.0->autogluon.core==0.3.1->autogluon) (1.2.0)\n",
            "Requirement already satisfied: toolz>=0.8.2 in /usr/local/lib/python3.7/dist-packages (from dask>=2.6.0->autogluon.core==0.3.1->autogluon) (0.11.2)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.7/dist-packages (from dask>=2.6.0->autogluon.core==0.3.1->autogluon) (21.3)\n",
            "Requirement already satisfied: msgpack>=0.6.0 in /usr/local/lib/python3.7/dist-packages (from distributed>=2.6.0->autogluon.core==0.3.1->autogluon) (1.0.3)\n",
            "Requirement already satisfied: tblib>=1.6.0 in /usr/local/lib/python3.7/dist-packages (from distributed>=2.6.0->autogluon.core==0.3.1->autogluon) (1.7.0)\n",
            "Requirement already satisfied: click>=6.6 in /usr/local/lib/python3.7/dist-packages (from distributed>=2.6.0->autogluon.core==0.3.1->autogluon) (7.1.2)\n",
            "Requirement already satisfied: zict>=0.1.3 in /usr/local/lib/python3.7/dist-packages (from distributed>=2.6.0->autogluon.core==0.3.1->autogluon) (2.0.0)\n",
            "Requirement already satisfied: setuptools in /usr/local/lib/python3.7/dist-packages (from distributed>=2.6.0->autogluon.core==0.3.1->autogluon) (59.6.0)\n",
            "Requirement already satisfied: sortedcontainers!=2.0.0,!=2.0.1 in /usr/local/lib/python3.7/dist-packages (from distributed>=2.6.0->autogluon.core==0.3.1->autogluon) (2.4.0)\n",
            "Requirement already satisfied: jinja2 in /usr/local/lib/python3.7/dist-packages (from distributed>=2.6.0->autogluon.core==0.3.1->autogluon) (2.11.3)\n",
            "Requirement already satisfied: pip in /usr/local/lib/python3.7/dist-packages (from fastai<3.0,>=2.3.1->autogluon.tabular[all]==0.3.1->autogluon) (21.3.1)\n",
            "Requirement already satisfied: fastprogress>=0.2.4 in /usr/local/lib/python3.7/dist-packages (from fastai<3.0,>=2.3.1->autogluon.tabular[all]==0.3.1->autogluon) (1.0.0)\n",
            "Requirement already satisfied: torchvision>=0.8.2 in /usr/local/lib/python3.7/dist-packages (from fastai<3.0,>=2.3.1->autogluon.tabular[all]==0.3.1->autogluon) (0.11.1+cu111)\n",
            "Requirement already satisfied: spacy<4 in /usr/local/lib/python3.7/dist-packages (from fastai<3.0,>=2.3.1->autogluon.tabular[all]==0.3.1->autogluon) (2.2.4)\n",
            "Requirement already satisfied: fastcore<1.4,>=1.3.22 in /usr/local/lib/python3.7/dist-packages (from fastai<3.0,>=2.3.1->autogluon.tabular[all]==0.3.1->autogluon) (1.3.27)\n",
            "Requirement already satisfied: fastdownload<2,>=0.0.5 in /usr/local/lib/python3.7/dist-packages (from fastai<3.0,>=2.3.1->autogluon.tabular[all]==0.3.1->autogluon) (0.0.5)\n",
            "Requirement already satisfied: portalocker in /usr/local/lib/python3.7/dist-packages (from gluoncv<0.10.5,>=0.10.4->autogluon.extra==0.3.1->autogluon) (2.3.2)\n",
            "Requirement already satisfied: opencv-python in /usr/local/lib/python3.7/dist-packages (from gluoncv<0.10.5,>=0.10.4->autogluon.extra==0.3.1->autogluon) (4.1.2.30)\n",
            "Requirement already satisfied: autocfg in /usr/local/lib/python3.7/dist-packages (from gluoncv<0.10.5,>=0.10.4->autogluon.extra==0.3.1->autogluon) (0.0.8)\n",
            "Requirement already satisfied: wheel in /usr/local/lib/python3.7/dist-packages (from lightgbm<4.0,>=3.0->autogluon.tabular[all]==0.3.1->autogluon) (0.37.0)\n",
            "Requirement already satisfied: python-dateutil>=2.7.3 in /usr/local/lib/python3.7/dist-packages (from pandas<2.0,>=1.0.0->autogluon.core==0.3.1->autogluon) (2.8.2)\n",
            "Requirement already satisfied: pytz>=2017.2 in /usr/local/lib/python3.7/dist-packages (from pandas<2.0,>=1.0.0->autogluon.core==0.3.1->autogluon) (2018.9)\n",
            "Requirement already satisfied: bcrypt>=3.1.3 in /usr/local/lib/python3.7/dist-packages (from paramiko>=2.4->autogluon.core==0.3.1->autogluon) (3.2.0)\n",
            "Requirement already satisfied: pynacl>=1.0.1 in /usr/local/lib/python3.7/dist-packages (from paramiko>=2.4->autogluon.core==0.3.1->autogluon) (1.4.0)\n",
            "Requirement already satisfied: cryptography>=2.5 in /usr/local/lib/python3.7/dist-packages (from paramiko>=2.4->autogluon.core==0.3.1->autogluon) (36.0.1)\n",
            "Requirement already satisfied: joblib>=0.11 in /usr/local/lib/python3.7/dist-packages (from scikit-learn<0.25,>=0.23.2->autogluon.core==0.3.1->autogluon) (1.1.0)\n",
            "Requirement already satisfied: threadpoolctl>=2.0.0 in /usr/local/lib/python3.7/dist-packages (from scikit-learn<0.25,>=0.23.2->autogluon.core==0.3.1->autogluon) (3.0.0)\n",
            "Requirement already satisfied: typing-extensions in /usr/local/lib/python3.7/dist-packages (from torch<2.0,>=1.0->autogluon.tabular[all]==0.3.1->autogluon) (3.10.0.2)\n",
            "Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in /usr/local/lib/python3.7/dist-packages (from boto3->autogluon.core==0.3.1->autogluon) (0.10.0)\n",
            "Requirement already satisfied: s3transfer<0.6.0,>=0.5.0 in /usr/local/lib/python3.7/dist-packages (from boto3->autogluon.core==0.3.1->autogluon) (0.5.0)\n",
            "Requirement already satisfied: botocore<1.24.0,>=1.23.24 in /usr/local/lib/python3.7/dist-packages (from boto3->autogluon.core==0.3.1->autogluon) (1.23.24)\n",
            "Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.7/dist-packages (from matplotlib->autogluon.core==0.3.1->autogluon) (1.3.2)\n",
            "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.7/dist-packages (from matplotlib->autogluon.core==0.3.1->autogluon) (0.11.0)\n",
            "Requirement already satisfied: xmltodict in /usr/local/lib/python3.7/dist-packages (from openml->autogluon.extra==0.3.1->autogluon) (0.12.0)\n",
            "Requirement already satisfied: liac-arff>=2.4.0 in /usr/local/lib/python3.7/dist-packages (from openml->autogluon.extra==0.3.1->autogluon) (2.5.0)\n",
            "Requirement already satisfied: minio in /usr/local/lib/python3.7/dist-packages (from openml->autogluon.extra==0.3.1->autogluon) (7.1.2)\n",
            "Requirement already satisfied: atomicwrites>=1.0 in /usr/local/lib/python3.7/dist-packages (from pytest->autogluon.extra==0.3.1->autogluon) (1.4.0)\n",
            "Requirement already satisfied: attrs>=17.4.0 in /usr/local/lib/python3.7/dist-packages (from pytest->autogluon.extra==0.3.1->autogluon) (21.2.0)\n",
            "Requirement already satisfied: pluggy<0.8,>=0.5 in /usr/local/lib/python3.7/dist-packages (from pytest->autogluon.extra==0.3.1->autogluon) (0.7.1)\n",
            "Requirement already satisfied: py>=1.5.0 in /usr/local/lib/python3.7/dist-packages (from pytest->autogluon.extra==0.3.1->autogluon) (1.11.0)\n",
            "Requirement already satisfied: more-itertools>=4.0.0 in /usr/local/lib/python3.7/dist-packages (from pytest->autogluon.extra==0.3.1->autogluon) (8.12.0)\n",
            "Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.7/dist-packages (from requests->autogluon.core==0.3.1->autogluon) (1.25.11)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.7/dist-packages (from requests->autogluon.core==0.3.1->autogluon) (2021.10.8)\n",
            "Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.7/dist-packages (from requests->autogluon.core==0.3.1->autogluon) (3.0.4)\n",
            "Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.7/dist-packages (from requests->autogluon.core==0.3.1->autogluon) (2.10)\n",
            "Requirement already satisfied: cffi>=1.1 in /usr/local/lib/python3.7/dist-packages (from bcrypt>=3.1.3->paramiko>=2.4->autogluon.core==0.3.1->autogluon) (1.15.0)\n",
            "Requirement already satisfied: locket in /usr/local/lib/python3.7/dist-packages (from partd>=0.3.10->dask>=2.6.0->autogluon.core==0.3.1->autogluon) (0.2.1)\n",
            "Requirement already satisfied: plac<1.2.0,>=0.9.6 in /usr/local/lib/python3.7/dist-packages (from spacy<4->fastai<3.0,>=2.3.1->autogluon.tabular[all]==0.3.1->autogluon) (1.1.3)\n",
            "Requirement already satisfied: murmurhash<1.1.0,>=0.28.0 in /usr/local/lib/python3.7/dist-packages (from spacy<4->fastai<3.0,>=2.3.1->autogluon.tabular[all]==0.3.1->autogluon) (1.0.6)\n",
            "Requirement already satisfied: wasabi<1.1.0,>=0.4.0 in /usr/local/lib/python3.7/dist-packages (from spacy<4->fastai<3.0,>=2.3.1->autogluon.tabular[all]==0.3.1->autogluon) (0.8.2)\n",
            "Requirement already satisfied: catalogue<1.1.0,>=0.0.7 in /usr/local/lib/python3.7/dist-packages (from spacy<4->fastai<3.0,>=2.3.1->autogluon.tabular[all]==0.3.1->autogluon) (1.0.0)\n",
            "Requirement already satisfied: cymem<2.1.0,>=2.0.2 in /usr/local/lib/python3.7/dist-packages (from spacy<4->fastai<3.0,>=2.3.1->autogluon.tabular[all]==0.3.1->autogluon) (2.0.6)\n",
            "Requirement already satisfied: blis<0.5.0,>=0.4.0 in /usr/local/lib/python3.7/dist-packages (from spacy<4->fastai<3.0,>=2.3.1->autogluon.tabular[all]==0.3.1->autogluon) (0.4.1)\n",
            "Requirement already satisfied: srsly<1.1.0,>=1.0.2 in /usr/local/lib/python3.7/dist-packages (from spacy<4->fastai<3.0,>=2.3.1->autogluon.tabular[all]==0.3.1->autogluon) (1.0.5)\n",
            "Requirement already satisfied: thinc==7.4.0 in /usr/local/lib/python3.7/dist-packages (from spacy<4->fastai<3.0,>=2.3.1->autogluon.tabular[all]==0.3.1->autogluon) (7.4.0)\n",
            "Requirement already satisfied: preshed<3.1.0,>=3.0.2 in /usr/local/lib/python3.7/dist-packages (from spacy<4->fastai<3.0,>=2.3.1->autogluon.tabular[all]==0.3.1->autogluon) (3.0.6)\n",
            "Requirement already satisfied: heapdict in /usr/local/lib/python3.7/dist-packages (from zict>=0.1.3->distributed>=2.6.0->autogluon.core==0.3.1->autogluon) (1.0.1)\n",
            "Requirement already satisfied: immutables>=0.9 in /usr/local/lib/python3.7/dist-packages (from contextvars->autogluon-contrib-nlp==0.0.1b20210201->autogluon.text==0.3.1->autogluon) (0.16)\n",
            "Requirement already satisfied: mccabe<0.7.0,>=0.6.0 in /usr/local/lib/python3.7/dist-packages (from flake8->autogluon-contrib-nlp==0.0.1b20210201->autogluon.text==0.3.1->autogluon) (0.6.1)\n",
            "Requirement already satisfied: pyflakes<2.5.0,>=2.4.0 in /usr/local/lib/python3.7/dist-packages (from flake8->autogluon-contrib-nlp==0.0.1b20210201->autogluon.text==0.3.1->autogluon) (2.4.0)\n",
            "Requirement already satisfied: pycodestyle<2.9.0,>=2.8.0 in /usr/local/lib/python3.7/dist-packages (from flake8->autogluon-contrib-nlp==0.0.1b20210201->autogluon.text==0.3.1->autogluon) (2.8.0)\n",
            "Requirement already satisfied: importlib-metadata<4.3 in /usr/local/lib/python3.7/dist-packages (from flake8->autogluon-contrib-nlp==0.0.1b20210201->autogluon.text==0.3.1->autogluon) (4.2.0)\n",
            "Requirement already satisfied: MarkupSafe>=0.23 in /usr/local/lib/python3.7/dist-packages (from jinja2->distributed>=2.6.0->autogluon.core==0.3.1->autogluon) (2.0.1)\n",
            "Requirement already satisfied: python-slugify in /usr/local/lib/python3.7/dist-packages (from kaggle->d8<1.0,>=0.0.2->autogluon.vision==0.3.1->autogluon) (5.0.2)\n",
            "Requirement already satisfied: retrying>=1.3.3 in /usr/local/lib/python3.7/dist-packages (from plotly->catboost<0.26,>=0.24.0->autogluon.tabular[all]==0.3.1->autogluon) (1.3.3)\n",
            "Requirement already satisfied: colorama in /usr/local/lib/python3.7/dist-packages (from sacrebleu->autogluon-contrib-nlp==0.0.1b20210201->autogluon.text==0.3.1->autogluon) (0.4.4)\n",
            "Requirement already satisfied: tabulate>=0.8.9 in /usr/local/lib/python3.7/dist-packages (from sacrebleu->autogluon-contrib-nlp==0.0.1b20210201->autogluon.text==0.3.1->autogluon) (0.8.9)\n",
            "Requirement already satisfied: pycparser in /usr/local/lib/python3.7/dist-packages (from cffi>=1.1->bcrypt>=3.1.3->paramiko>=2.4->autogluon.core==0.3.1->autogluon) (2.21)\n",
            "Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.7/dist-packages (from importlib-metadata<4.3->flake8->autogluon-contrib-nlp==0.0.1b20210201->autogluon.text==0.3.1->autogluon) (3.6.0)\n",
            "Requirement already satisfied: text-unidecode>=1.3 in /usr/local/lib/python3.7/dist-packages (from python-slugify->kaggle->d8<1.0,>=0.0.2->autogluon.vision==0.3.1->autogluon) (1.3)\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\n"
          ]
        }
      ],
      "source": [
        "#@title Install prerequsite: you may have to do Runtime -> Restart runtime after the installation\n",
        "\n",
        "!pip install --upgrade pip\n",
        "!pip install --upgrade setuptools wheel\n",
        "!pip install --upgrade \"mxnet<2.0.0\"\n",
        "!pip install autogluon"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "A8VaE4FjdJSf"
      },
      "outputs": [],
      "source": [
        "import autogluon"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4_mQlXl4BnDA",
        "outputId": "9d7a65b6-5728-439c-8ab1-1b85d0fe916a"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n"
          ]
        }
      ],
      "source": [
        "#@title Mount Google Drive (You don't need to run this if you are running notebooks on your laptop)\n",
        "\n",
        "from google.colab import drive\n",
        "\n",
        "# The following command will prompt a URL for you to click and obtain the\n",
        "# authorization code\n",
        "\n",
        "drive.mount(\"/content/drive\")\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "QQD4mkx-CCmA"
      },
      "outputs": [],
      "source": [
        "# Set up data folder\n",
        "from pathlib import Path\n",
        "\n",
        "# Change this to where you put your hw2 files\n",
        "DATA = Path(\"/content/drive/My Drive/project\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "QEpLQx3tR08M"
      },
      "source": [
        "# Project part 1 "
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "WVpMokdnR08U"
      },
      "source": [
        "## Reclassifying Mutations\n",
        "\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "sCIkTJEDlwnN"
      },
      "source": [
        "First the existing mutation files from the paper were loaded. The mutations from the files were combined a kept seperate by gene. The total mutations for each type of gene found in this analysis match the mutations statistics record in the meta files from the paper. "
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "uREv53yKP8s1"
      },
      "outputs": [],
      "source": [
        "import pandas as pd\n",
        "snvsfile = DATA / \"snvs.txt\"\n",
        "snvs = pd.read_csv(snvsfile, sep=\"\\t\", comment=\"#\")\n",
        "snvs = snvs[['Gene', 'Sample_ID']]\n",
        "snvs = snvs.dropna()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "7D5uPaoiWLXi"
      },
      "outputs": [],
      "source": [
        "import pandas as pd\n",
        "indelsfile = DATA / \"indels.txt\"\n",
        "indels = pd.read_csv(indelsfile, sep=\"\\t\", comment=\"#\")\n",
        "indels = indels[['Gene', 'Sample_ID']]\n",
        "indels = indels.dropna()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 238
        },
        "id": "bC9rCzvRYslA",
        "outputId": "0c222139-310c-40a1-8f6e-8d18a6b2ca74"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Sample_ID</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Gene</th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>SMARCB1</th>\n",
              "      <td>5</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>KRT39</th>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>MUC16</th>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>OR4C6</th>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ARMCX4</th>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "         Sample_ID\n",
              "Gene              \n",
              "SMARCB1          5\n",
              "KRT39            2\n",
              "MUC16            2\n",
              "OR4C6            2\n",
              "ARMCX4           2"
            ]
          },
          "execution_count": 7,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "groupedsnvs = snvs.groupby(by=['Gene']).count()\n",
        "groupedsnvs = groupedsnvs.sort_values(by = [\"Sample_ID\"], ascending=False)\n",
        "groupedsnvs.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 238
        },
        "id": "Xbli1UOFWnWg",
        "outputId": "063a5a56-8925-4ffa-be98-f309c6861481"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Sample_ID</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Gene</th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>SMARCB1</th>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ANK2</th>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>AUTS2</th>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ZBTB48</th>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>TBX15</th>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "         Sample_ID\n",
              "Gene              \n",
              "SMARCB1          2\n",
              "ANK2             1\n",
              "AUTS2            1\n",
              "ZBTB48           1\n",
              "TBX15            1"
            ]
          },
          "execution_count": 8,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "groupedindels = indels.groupby(by=['Gene']).count()\n",
        "groupedindels = groupedindels.sort_values(by = [\"Sample_ID\"], ascending=False)\n",
        "groupedindels.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "GC4OmvWS1qbC"
      },
      "outputs": [],
      "source": [
        "concat = pd.concat([snvs, indels])"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 238
        },
        "id": "hYviEMnE7dH7",
        "outputId": "25c55a12-0d0b-45bf-d7d4-9146c9ebcfc6"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>unique</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Gene</th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>SMARCB1</th>\n",
              "      <td>7</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>MUC16</th>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ARMCX4</th>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>PPARGC1A</th>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>RAB11FIP2</th>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "          unique\n",
              "Gene            \n",
              "SMARCB1        7\n",
              "MUC16          2\n",
              "ARMCX4         2\n",
              "PPARGC1A       1\n",
              "RAB11FIP2      1"
            ]
          },
          "execution_count": 10,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "count = concat.groupby(by='Gene')\n",
        "count = count.describe()\n",
        "count = count['Sample_ID']\n",
        "count = count.sort_values(by ='unique', ascending=False)\n",
        "count = count[['unique']]\n",
        "count.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 455
        },
        "id": "XXDYGCF4xezr",
        "outputId": "750f2b88-65fa-453b-c659-18be0dd4f679"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>unique</th>\n",
              "      <th>snvs</th>\n",
              "      <th>indel</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Gene</th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>SMARCB1</th>\n",
              "      <td>7</td>\n",
              "      <td>5</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>MUC16</th>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ARMCX4</th>\n",
              "      <td>2</td>\n",
              "      <td>2</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>PPARGC1A</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>RAB11FIP2</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>IFIT3</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>IL36B</th>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ISL2</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ITGB4</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ZNF791</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>248 rows × 3 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "           unique  snvs  indel\n",
              "Gene                          \n",
              "SMARCB1         7     5      2\n",
              "MUC16           2     2      0\n",
              "ARMCX4          2     2      0\n",
              "PPARGC1A        1     1      0\n",
              "RAB11FIP2       1     1      0\n",
              "...           ...   ...    ...\n",
              "IFIT3           1     1      0\n",
              "IL36B           1     0      1\n",
              "ISL2            1     1      0\n",
              "ITGB4           1     1      0\n",
              "ZNF791          1     1      0\n",
              "\n",
              "[248 rows x 3 columns]"
            ]
          },
          "execution_count": 11,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "count['snvs'] = groupedsnvs.Sample_ID\n",
        "count['indel'] = groupedindels.Sample_ID\n",
        "summary = count.fillna(0)\n",
        "summary = summary.astype(int)\n",
        "summary"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Project Part 2"
      ],
      "metadata": {
        "id": "D3jea3v_54LV"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Machine Learning"
      ],
      "metadata": {
        "id": "iqZZb03P6Bq_"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "The paper uses methylation data to divide the atrt tumors into three seperate subcatagories. The gene expression data contains similar data but unlike the methylation data it is not specific to a location. We started by loading the gene expression and methylation data for 49 cases from the 180 atrt patients in the paper. The subtype labels from the paper are then matched with these 49 samples by sample name. The 49 samples are now split into 3 groups and can be used in a multiclass training method by autogluon. The data was split into 40 training cases and 9 test cases, 3 from each class. "
      ],
      "metadata": {
        "id": "Pxd5vsOb6mWM"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8TCbmVGlXQNa",
        "outputId": "28e4833f-39fe-457f-a2e8-63aaa0322515"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2718: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49) have mixed types.Specify dtype option on import or set low_memory=False.\n",
            "  interactivity=interactivity, compiler=compiler, result=result)\n"
          ]
        }
      ],
      "source": [
        "import pandas as pd\n",
        "\n",
        "gefile = DATA / \"GSE70678_matrix.txt\"\n",
        "methfile = DATA / \"GSE70460_series_matrix.txt.gz\"\n",
        "\n",
        "ge = pd.read_csv(gefile, sep=\"\\t\", na_values=\"unknown\", index_col=0)\n",
        "meth = pd.read_csv(methfile, sep=\"\\t\", na_values=\"unknown\", index_col=0, header= 64)\n",
        "gemeta = ge.head(36)\n",
        "ge.columns = ge.iloc[35]\n",
        "ge = ge.iloc[36:-1 ,:]"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "r0WoL9dLWkrf",
        "outputId": "d0804e94-c5c7-407d-bdac-68dc39d388e3"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>dkfz_ATRT_159</th>\n",
              "      <th>dkfz_ATRT_17</th>\n",
              "      <th>dkfz_ATRT_25</th>\n",
              "      <th>dkfz_ATRT_173</th>\n",
              "      <th>dkfz_ATRT_151</th>\n",
              "      <th>dkfz_ATRT_171</th>\n",
              "      <th>dkfz_ATRT_152</th>\n",
              "      <th>dkfz_ATRT_172</th>\n",
              "      <th>dkfz_ATRT_153</th>\n",
              "      <th>dkfz_ATRT_86</th>\n",
              "      <th>dkfz_ATRT_10</th>\n",
              "      <th>dkfz_ATRT_154</th>\n",
              "      <th>dkfz_ATRT_30</th>\n",
              "      <th>dkfz_ATRT_36</th>\n",
              "      <th>dkfz_ATRT_11</th>\n",
              "      <th>dkfz_ATRT_12</th>\n",
              "      <th>dkfz_ATRT_41</th>\n",
              "      <th>dkfz_ATRT_43</th>\n",
              "      <th>dkfz_ATRT_164</th>\n",
              "      <th>dkfz_ATRT_16</th>\n",
              "      <th>dkfz_ATRT_6</th>\n",
              "      <th>dkfz_ATRT_160</th>\n",
              "      <th>dkfz_ATRT_7</th>\n",
              "      <th>dkfz_ATRT_48</th>\n",
              "      <th>dkfz_ATRT_49</th>\n",
              "      <th>dkfz_ATRT_51</th>\n",
              "      <th>dkfz_ATRT_47</th>\n",
              "      <th>dkfz_ATRT_46</th>\n",
              "      <th>dkfz_ATRT_101</th>\n",
              "      <th>dkfz_ATRT_157</th>\n",
              "      <th>dkfz_ATRT_155</th>\n",
              "      <th>dkfz_ATRT_168</th>\n",
              "      <th>dkfz_ATRT_156</th>\n",
              "      <th>dkfz_ATRT_163</th>\n",
              "      <th>dkfz_ATRT_162</th>\n",
              "      <th>dkfz_ATRT_158</th>\n",
              "      <th>dkfz_ATRT_166</th>\n",
              "      <th>dkfz_ATRT_169</th>\n",
              "      <th>dkfz_ATRT_170</th>\n",
              "      <th>dkfz_ATRT_88</th>\n",
              "      <th>dkfz_ATRT_72</th>\n",
              "      <th>dkfz_ATRT_167</th>\n",
              "      <th>dkfz_ATRT_85</th>\n",
              "      <th>dkfz_ATRT_84</th>\n",
              "      <th>dkfz_ATRT_165</th>\n",
              "      <th>dkfz_ATRT_14</th>\n",
              "      <th>dkfz_ATRT_32</th>\n",
              "      <th>dkfz_ATRT_161</th>\n",
              "      <th>dkfz_ATRT_90</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_title</th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>!Sample_geo_accession</th>\n",
              "      <td>GSM1816332</td>\n",
              "      <td>GSM1816333</td>\n",
              "      <td>GSM1816334</td>\n",
              "      <td>GSM1816335</td>\n",
              "      <td>GSM1816336</td>\n",
              "      <td>GSM1816337</td>\n",
              "      <td>GSM1816338</td>\n",
              "      <td>GSM1816339</td>\n",
              "      <td>GSM1816340</td>\n",
              "      <td>GSM1816341</td>\n",
              "      <td>GSM1816342</td>\n",
              "      <td>GSM1816343</td>\n",
              "      <td>GSM1816344</td>\n",
              "      <td>GSM1816345</td>\n",
              "      <td>GSM1816346</td>\n",
              "      <td>GSM1816347</td>\n",
              "      <td>GSM1816348</td>\n",
              "      <td>GSM1816349</td>\n",
              "      <td>GSM1816350</td>\n",
              "      <td>GSM1816351</td>\n",
              "      <td>GSM1816352</td>\n",
              "      <td>GSM1816353</td>\n",
              "      <td>GSM1816354</td>\n",
              "      <td>GSM1816355</td>\n",
              "      <td>GSM1816356</td>\n",
              "      <td>GSM1816357</td>\n",
              "      <td>GSM1816358</td>\n",
              "      <td>GSM1816359</td>\n",
              "      <td>GSM1816360</td>\n",
              "      <td>GSM1816361</td>\n",
              "      <td>GSM1816362</td>\n",
              "      <td>GSM1816363</td>\n",
              "      <td>GSM1816364</td>\n",
              "      <td>GSM1816365</td>\n",
              "      <td>GSM1816366</td>\n",
              "      <td>GSM1816367</td>\n",
              "      <td>GSM1816368</td>\n",
              "      <td>GSM1816369</td>\n",
              "      <td>GSM1816370</td>\n",
              "      <td>GSM1816371</td>\n",
              "      <td>GSM1816372</td>\n",
              "      <td>GSM1816373</td>\n",
              "      <td>GSM1816374</td>\n",
              "      <td>GSM1816375</td>\n",
              "      <td>GSM1816376</td>\n",
              "      <td>GSM1816377</td>\n",
              "      <td>GSM1816378</td>\n",
              "      <td>GSM1816379</td>\n",
              "      <td>GSM1816380</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_status</th>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_submission_date</th>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_last_update_date</th>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_type</th>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "      <td>RNA</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_channel_count</th>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_source_name_ch1</th>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_organism_ch1</th>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>Homo sapiens</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_characteristics_ch1</th>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: NA</td>\n",
              "      <td>gender: NA</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>gender: M</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_characteristics_ch1</th>\n",
              "      <td>age (years): 1</td>\n",
              "      <td>age (years): 2</td>\n",
              "      <td>age (years): 1.3</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): 0.3</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): 2</td>\n",
              "      <td>age (years): 1</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): 1</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): 0</td>\n",
              "      <td>age (years): 1</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): 1</td>\n",
              "      <td>age (years): 1.5</td>\n",
              "      <td>age (years): 1.5</td>\n",
              "      <td>age (years): 9.5</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): 4</td>\n",
              "      <td>age (years): 2.9</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): 1.3</td>\n",
              "      <td>age (years): 1.5</td>\n",
              "      <td>age (years): 2.8</td>\n",
              "      <td>age (years): 1.5</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): 4.3</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>age (years): 1.5</td>\n",
              "      <td>age (years): 2</td>\n",
              "      <td>age (years): 0.5</td>\n",
              "      <td>age (years): 3</td>\n",
              "      <td>age (years): 0.2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_characteristics_ch1</th>\n",
              "      <td>tumor location: Infratentorial</td>\n",
              "      <td>tumor location: Supratentorial</td>\n",
              "      <td>tumor location: Infratentorial</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: Infratentorial</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: Infratentorial</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: Infratentorial</td>\n",
              "      <td>tumor location: Infratentorial</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: Supratentorial</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: Infratentorial</td>\n",
              "      <td>tumor location: Infratentorial</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: Infratentorial</td>\n",
              "      <td>tumor location: Supratentorial</td>\n",
              "      <td>tumor location: Infratentorial</td>\n",
              "      <td>tumor location: Supratentorial</td>\n",
              "      <td>tumor location: Supratentorial</td>\n",
              "      <td>tumor location: Infratentorial</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: Supratentorial</td>\n",
              "      <td>tumor location: Infratentorial</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: Supratentorial</td>\n",
              "      <td>tumor location: Supratentorial</td>\n",
              "      <td>tumor location: Supratentorial</td>\n",
              "      <td>tumor location: Infratentorial</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: Infratentorial</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>tumor location: Infratentorial</td>\n",
              "      <td>tumor location: Infratentorial</td>\n",
              "      <td>tumor location: Supratentorial</td>\n",
              "      <td>tumor location: Infratentorial</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_treatment_protocol_ch1</th>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_growth_protocol_ch1</th>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_molecule_ch1</th>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>total RNA</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_extract_protocol_ch1</th>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_label_ch1</th>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "      <td>biotin</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_label_protocol_ch1</th>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_taxid_ch1</th>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "      <td>9606</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_hyb_protocol</th>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_scan_protocol</th>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_description</th>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_data_processing</th>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_platform_id</th>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>GPL570</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_contact_name</th>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_contact_email</th>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_contact_department</th>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_contact_institute</th>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_contact_address</th>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_contact_city</th>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>Heidelberg</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_contact_zip/postal_code</th>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "      <td>69120</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_contact_country</th>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "      <td>Germany</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_supplementary_file</th>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_data_row_count</th>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "      <td>54675</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_relation</th>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!series_matrix_table_begin</th>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ID_REF</th>\n",
              "      <td>GSM1816332</td>\n",
              "      <td>GSM1816333</td>\n",
              "      <td>GSM1816334</td>\n",
              "      <td>GSM1816335</td>\n",
              "      <td>GSM1816336</td>\n",
              "      <td>GSM1816337</td>\n",
              "      <td>GSM1816338</td>\n",
              "      <td>GSM1816339</td>\n",
              "      <td>GSM1816340</td>\n",
              "      <td>GSM1816341</td>\n",
              "      <td>GSM1816342</td>\n",
              "      <td>GSM1816343</td>\n",
              "      <td>GSM1816344</td>\n",
              "      <td>GSM1816345</td>\n",
              "      <td>GSM1816346</td>\n",
              "      <td>GSM1816347</td>\n",
              "      <td>GSM1816348</td>\n",
              "      <td>GSM1816349</td>\n",
              "      <td>GSM1816350</td>\n",
              "      <td>GSM1816351</td>\n",
              "      <td>GSM1816352</td>\n",
              "      <td>GSM1816353</td>\n",
              "      <td>GSM1816354</td>\n",
              "      <td>GSM1816355</td>\n",
              "      <td>GSM1816356</td>\n",
              "      <td>GSM1816357</td>\n",
              "      <td>GSM1816358</td>\n",
              "      <td>GSM1816359</td>\n",
              "      <td>GSM1816360</td>\n",
              "      <td>GSM1816361</td>\n",
              "      <td>GSM1816362</td>\n",
              "      <td>GSM1816363</td>\n",
              "      <td>GSM1816364</td>\n",
              "      <td>GSM1816365</td>\n",
              "      <td>GSM1816366</td>\n",
              "      <td>GSM1816367</td>\n",
              "      <td>GSM1816368</td>\n",
              "      <td>GSM1816369</td>\n",
              "      <td>GSM1816370</td>\n",
              "      <td>GSM1816371</td>\n",
              "      <td>GSM1816372</td>\n",
              "      <td>GSM1816373</td>\n",
              "      <td>GSM1816374</td>\n",
              "      <td>GSM1816375</td>\n",
              "      <td>GSM1816376</td>\n",
              "      <td>GSM1816377</td>\n",
              "      <td>GSM1816378</td>\n",
              "      <td>GSM1816379</td>\n",
              "      <td>GSM1816380</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "                                                                     dkfz_ATRT_159  ...                                       dkfz_ATRT_90\n",
              "!Sample_title                                                                       ...                                                   \n",
              "!Sample_geo_accession                                                   GSM1816332  ...                                         GSM1816380\n",
              "!Sample_status                                               Public on Mar 02 2016  ...                              Public on Mar 02 2016\n",
              "!Sample_submission_date                                                Jul 09 2015  ...                                        Jul 09 2015\n",
              "!Sample_last_update_date                                               Mar 13 2018  ...                                        Mar 13 2018\n",
              "!Sample_type                                                                   RNA  ...                                                RNA\n",
              "!Sample_channel_count                                                            1  ...                                                  1\n",
              "!Sample_source_name_ch1                                  Primary ATRT tumor tissue  ...                          Primary ATRT tumor tissue\n",
              "!Sample_organism_ch1                                                  Homo sapiens  ...                                       Homo sapiens\n",
              "!Sample_characteristics_ch1                                              gender: M  ...                                          gender: M\n",
              "!Sample_characteristics_ch1                                         age (years): 1  ...                                   age (years): 0.2\n",
              "!Sample_characteristics_ch1                         tumor location: Infratentorial  ...                     tumor location: Infratentorial\n",
              "!Sample_treatment_protocol_ch1                                                 NaN  ...                                                NaN\n",
              "!Sample_growth_protocol_ch1                                                    NaN  ...                                                NaN\n",
              "!Sample_molecule_ch1                                                     total RNA  ...                                          total RNA\n",
              "!Sample_extract_protocol_ch1     Total RNA from biopsy samples was isolated aft...  ...  Total RNA from biopsy samples was isolated aft...\n",
              "!Sample_label_ch1                                                           biotin  ...                                             biotin\n",
              "!Sample_label_protocol_ch1                                     Affymetrix protocol  ...                                Affymetrix protocol\n",
              "!Sample_taxid_ch1                                                             9606  ...                                               9606\n",
              "!Sample_hyb_protocol                                           Affymetrix protocol  ...                                Affymetrix protocol\n",
              "!Sample_scan_protocol                                          Affymetrix protocol  ...                                Affymetrix protocol\n",
              "!Sample_description                 Gene expression data from an ATRT tumor sample  ...     Gene expression data from an ATRT tumor sample\n",
              "!Sample_data_processing          For normalization of the expression data we us...  ...  For normalization of the expression data we us...\n",
              "!Sample_platform_id                                                         GPL570  ...                                             GPL570\n",
              "!Sample_contact_name                                                  Marcel,,Kool  ...                                       Marcel,,Kool\n",
              "!Sample_contact_email                                               m.kool@dkfz.de  ...                                     m.kool@dkfz.de\n",
              "!Sample_contact_department                                 Pediatric Neurooncology  ...                            Pediatric Neurooncology\n",
              "!Sample_contact_institute                       German Cancer Research Center DKFZ  ...                 German Cancer Research Center DKFZ\n",
              "!Sample_contact_address                                    Im Neuenheimer Feld 280  ...                            Im Neuenheimer Feld 280\n",
              "!Sample_contact_city                                                    Heidelberg  ...                                         Heidelberg\n",
              "!Sample_contact_zip/postal_code                                              69120  ...                                              69120\n",
              "!Sample_contact_country                                                    Germany  ...                                            Germany\n",
              "!Sample_supplementary_file       ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...  ...  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...\n",
              "!Sample_data_row_count                                                       54675  ...                                              54675\n",
              "!Sample_relation                                          Reanalyzed by: GSE111787  ...                           Reanalyzed by: GSE111787\n",
              "!series_matrix_table_begin                                                     NaN  ...                                                NaN\n",
              "ID_REF                                                                  GSM1816332  ...                                         GSM1816380\n",
              "\n",
              "[36 rows x 49 columns]"
            ]
          },
          "execution_count": 13,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "gemeta.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 258
        },
        "id": "M_-WRexvW3Bw",
        "outputId": "4d0cd548-0e4f-44e7-a38d-0f4f8ea6382e"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th>ID_REF</th>\n",
              "      <th>GSM1816332</th>\n",
              "      <th>GSM1816333</th>\n",
              "      <th>GSM1816334</th>\n",
              "      <th>GSM1816335</th>\n",
              "      <th>GSM1816336</th>\n",
              "      <th>GSM1816337</th>\n",
              "      <th>GSM1816338</th>\n",
              "      <th>GSM1816339</th>\n",
              "      <th>GSM1816340</th>\n",
              "      <th>GSM1816341</th>\n",
              "      <th>GSM1816342</th>\n",
              "      <th>GSM1816343</th>\n",
              "      <th>GSM1816344</th>\n",
              "      <th>GSM1816345</th>\n",
              "      <th>GSM1816346</th>\n",
              "      <th>GSM1816347</th>\n",
              "      <th>GSM1816348</th>\n",
              "      <th>GSM1816349</th>\n",
              "      <th>GSM1816350</th>\n",
              "      <th>GSM1816351</th>\n",
              "      <th>GSM1816352</th>\n",
              "      <th>GSM1816353</th>\n",
              "      <th>GSM1816354</th>\n",
              "      <th>GSM1816355</th>\n",
              "      <th>GSM1816356</th>\n",
              "      <th>GSM1816357</th>\n",
              "      <th>GSM1816358</th>\n",
              "      <th>GSM1816359</th>\n",
              "      <th>GSM1816360</th>\n",
              "      <th>GSM1816361</th>\n",
              "      <th>GSM1816362</th>\n",
              "      <th>GSM1816363</th>\n",
              "      <th>GSM1816364</th>\n",
              "      <th>GSM1816365</th>\n",
              "      <th>GSM1816366</th>\n",
              "      <th>GSM1816367</th>\n",
              "      <th>GSM1816368</th>\n",
              "      <th>GSM1816369</th>\n",
              "      <th>GSM1816370</th>\n",
              "      <th>GSM1816371</th>\n",
              "      <th>GSM1816372</th>\n",
              "      <th>GSM1816373</th>\n",
              "      <th>GSM1816374</th>\n",
              "      <th>GSM1816375</th>\n",
              "      <th>GSM1816376</th>\n",
              "      <th>GSM1816377</th>\n",
              "      <th>GSM1816378</th>\n",
              "      <th>GSM1816379</th>\n",
              "      <th>GSM1816380</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>!Sample_title</th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>1007_s_at</th>\n",
              "      <td>426.5</td>\n",
              "      <td>1448.6</td>\n",
              "      <td>573.1</td>\n",
              "      <td>417.5</td>\n",
              "      <td>479.6</td>\n",
              "      <td>1243.2</td>\n",
              "      <td>529.4</td>\n",
              "      <td>1151.6</td>\n",
              "      <td>1126.9</td>\n",
              "      <td>588.7</td>\n",
              "      <td>669.3</td>\n",
              "      <td>112</td>\n",
              "      <td>1207.1</td>\n",
              "      <td>641</td>\n",
              "      <td>767.2</td>\n",
              "      <td>979.7</td>\n",
              "      <td>1199.5</td>\n",
              "      <td>1098.8</td>\n",
              "      <td>378.1</td>\n",
              "      <td>416.6</td>\n",
              "      <td>590.6</td>\n",
              "      <td>826.1</td>\n",
              "      <td>1116.2</td>\n",
              "      <td>873.6</td>\n",
              "      <td>671.6</td>\n",
              "      <td>536.3</td>\n",
              "      <td>114.2</td>\n",
              "      <td>2334.8</td>\n",
              "      <td>766.3</td>\n",
              "      <td>1193.7</td>\n",
              "      <td>637.7</td>\n",
              "      <td>686.2</td>\n",
              "      <td>313</td>\n",
              "      <td>455</td>\n",
              "      <td>217</td>\n",
              "      <td>1060.9</td>\n",
              "      <td>425.1</td>\n",
              "      <td>146.1</td>\n",
              "      <td>334.2</td>\n",
              "      <td>350.8</td>\n",
              "      <td>687.5</td>\n",
              "      <td>845.7</td>\n",
              "      <td>590.6</td>\n",
              "      <td>348.7</td>\n",
              "      <td>1382.3</td>\n",
              "      <td>567.2</td>\n",
              "      <td>1284.5</td>\n",
              "      <td>1014.9</td>\n",
              "      <td>605.3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1053_at</th>\n",
              "      <td>95.9</td>\n",
              "      <td>173.3</td>\n",
              "      <td>75.7</td>\n",
              "      <td>57.7</td>\n",
              "      <td>160.9</td>\n",
              "      <td>141</td>\n",
              "      <td>188.5</td>\n",
              "      <td>126.1</td>\n",
              "      <td>122.6</td>\n",
              "      <td>107.8</td>\n",
              "      <td>163.9</td>\n",
              "      <td>120.6</td>\n",
              "      <td>118.9</td>\n",
              "      <td>177.4</td>\n",
              "      <td>154.9</td>\n",
              "      <td>112.3</td>\n",
              "      <td>97.9</td>\n",
              "      <td>120.3</td>\n",
              "      <td>181.9</td>\n",
              "      <td>156.3</td>\n",
              "      <td>100</td>\n",
              "      <td>125.5</td>\n",
              "      <td>154.1</td>\n",
              "      <td>155.1</td>\n",
              "      <td>103.2</td>\n",
              "      <td>180.1</td>\n",
              "      <td>236.3</td>\n",
              "      <td>81</td>\n",
              "      <td>112.4</td>\n",
              "      <td>154.9</td>\n",
              "      <td>141.2</td>\n",
              "      <td>230.8</td>\n",
              "      <td>127.3</td>\n",
              "      <td>158.8</td>\n",
              "      <td>181.2</td>\n",
              "      <td>187.2</td>\n",
              "      <td>140.2</td>\n",
              "      <td>132.6</td>\n",
              "      <td>81.8</td>\n",
              "      <td>147.8</td>\n",
              "      <td>75.3</td>\n",
              "      <td>209.3</td>\n",
              "      <td>197.1</td>\n",
              "      <td>189.5</td>\n",
              "      <td>86.5</td>\n",
              "      <td>171.7</td>\n",
              "      <td>145.9</td>\n",
              "      <td>157</td>\n",
              "      <td>168.4</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>117_at</th>\n",
              "      <td>28.3</td>\n",
              "      <td>18.6</td>\n",
              "      <td>42.7</td>\n",
              "      <td>169.2</td>\n",
              "      <td>15.7</td>\n",
              "      <td>17.2</td>\n",
              "      <td>21.4</td>\n",
              "      <td>70.2</td>\n",
              "      <td>54.4</td>\n",
              "      <td>795.3</td>\n",
              "      <td>30.9</td>\n",
              "      <td>26.6</td>\n",
              "      <td>12.8</td>\n",
              "      <td>17</td>\n",
              "      <td>17.2</td>\n",
              "      <td>38</td>\n",
              "      <td>28.4</td>\n",
              "      <td>95.8</td>\n",
              "      <td>74.8</td>\n",
              "      <td>49.3</td>\n",
              "      <td>44</td>\n",
              "      <td>198.2</td>\n",
              "      <td>23.1</td>\n",
              "      <td>34.8</td>\n",
              "      <td>37.4</td>\n",
              "      <td>12.6</td>\n",
              "      <td>83.5</td>\n",
              "      <td>43.7</td>\n",
              "      <td>75.1</td>\n",
              "      <td>23.5</td>\n",
              "      <td>22.5</td>\n",
              "      <td>27.1</td>\n",
              "      <td>37.7</td>\n",
              "      <td>46.3</td>\n",
              "      <td>26.9</td>\n",
              "      <td>9.7</td>\n",
              "      <td>65.4</td>\n",
              "      <td>69.2</td>\n",
              "      <td>34.3</td>\n",
              "      <td>89.5</td>\n",
              "      <td>28.9</td>\n",
              "      <td>18.5</td>\n",
              "      <td>31.7</td>\n",
              "      <td>21.9</td>\n",
              "      <td>102.9</td>\n",
              "      <td>49.7</td>\n",
              "      <td>1333.4</td>\n",
              "      <td>18.8</td>\n",
              "      <td>63</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>121_at</th>\n",
              "      <td>94.3</td>\n",
              "      <td>45.3</td>\n",
              "      <td>57.7</td>\n",
              "      <td>72</td>\n",
              "      <td>55.1</td>\n",
              "      <td>63.3</td>\n",
              "      <td>80.4</td>\n",
              "      <td>36.3</td>\n",
              "      <td>68.6</td>\n",
              "      <td>91</td>\n",
              "      <td>95.9</td>\n",
              "      <td>133.8</td>\n",
              "      <td>134.6</td>\n",
              "      <td>100</td>\n",
              "      <td>69.5</td>\n",
              "      <td>91.4</td>\n",
              "      <td>79.2</td>\n",
              "      <td>55.5</td>\n",
              "      <td>71.9</td>\n",
              "      <td>61.6</td>\n",
              "      <td>61.7</td>\n",
              "      <td>98.8</td>\n",
              "      <td>52.1</td>\n",
              "      <td>92.9</td>\n",
              "      <td>140.6</td>\n",
              "      <td>125.4</td>\n",
              "      <td>86.2</td>\n",
              "      <td>61.4</td>\n",
              "      <td>61.2</td>\n",
              "      <td>120.5</td>\n",
              "      <td>72.6</td>\n",
              "      <td>78</td>\n",
              "      <td>88.2</td>\n",
              "      <td>93.9</td>\n",
              "      <td>64.1</td>\n",
              "      <td>66.9</td>\n",
              "      <td>101.1</td>\n",
              "      <td>124.1</td>\n",
              "      <td>143.5</td>\n",
              "      <td>101.4</td>\n",
              "      <td>77.6</td>\n",
              "      <td>121.2</td>\n",
              "      <td>80.9</td>\n",
              "      <td>68.3</td>\n",
              "      <td>132</td>\n",
              "      <td>118.7</td>\n",
              "      <td>64.3</td>\n",
              "      <td>60.2</td>\n",
              "      <td>77.8</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1255_g_at</th>\n",
              "      <td>2.5</td>\n",
              "      <td>0.7</td>\n",
              "      <td>5.7</td>\n",
              "      <td>58.4</td>\n",
              "      <td>3.1</td>\n",
              "      <td>2.4</td>\n",
              "      <td>3.2</td>\n",
              "      <td>10.5</td>\n",
              "      <td>7.3</td>\n",
              "      <td>7.5</td>\n",
              "      <td>6.8</td>\n",
              "      <td>1.1</td>\n",
              "      <td>5.2</td>\n",
              "      <td>6.9</td>\n",
              "      <td>3.5</td>\n",
              "      <td>3.4</td>\n",
              "      <td>1.4</td>\n",
              "      <td>10.9</td>\n",
              "      <td>12.5</td>\n",
              "      <td>18.6</td>\n",
              "      <td>2.4</td>\n",
              "      <td>21.5</td>\n",
              "      <td>0.8</td>\n",
              "      <td>7.1</td>\n",
              "      <td>8.7</td>\n",
              "      <td>4</td>\n",
              "      <td>7.8</td>\n",
              "      <td>6.8</td>\n",
              "      <td>1.9</td>\n",
              "      <td>8.4</td>\n",
              "      <td>11.2</td>\n",
              "      <td>4.4</td>\n",
              "      <td>4.5</td>\n",
              "      <td>8.2</td>\n",
              "      <td>1.5</td>\n",
              "      <td>6.5</td>\n",
              "      <td>2.4</td>\n",
              "      <td>9.5</td>\n",
              "      <td>4.2</td>\n",
              "      <td>7.4</td>\n",
              "      <td>8.3</td>\n",
              "      <td>8.4</td>\n",
              "      <td>4.3</td>\n",
              "      <td>1.8</td>\n",
              "      <td>6.9</td>\n",
              "      <td>1.5</td>\n",
              "      <td>1.3</td>\n",
              "      <td>37.8</td>\n",
              "      <td>9.2</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "ID_REF        GSM1816332 GSM1816333 GSM1816334  ... GSM1816378 GSM1816379 GSM1816380\n",
              "!Sample_title                                   ...                                 \n",
              "1007_s_at          426.5     1448.6      573.1  ...     1284.5     1014.9      605.3\n",
              "1053_at             95.9      173.3       75.7  ...      145.9        157      168.4\n",
              "117_at              28.3       18.6       42.7  ...     1333.4       18.8         63\n",
              "121_at              94.3       45.3       57.7  ...       64.3       60.2       77.8\n",
              "1255_g_at            2.5        0.7        5.7  ...        1.3       37.8        9.2\n",
              "\n",
              "[5 rows x 49 columns]"
            ]
          },
          "execution_count": 14,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "ge.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 287
        },
        "id": "Q6qfjOMjOJZQ",
        "outputId": "f34fa9e8-ad31-4478-eb71-90d9ea71442f"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>GSM1755170</th>\n",
              "      <th>GSM1755171</th>\n",
              "      <th>GSM1755172</th>\n",
              "      <th>GSM1755173</th>\n",
              "      <th>GSM1755174</th>\n",
              "      <th>GSM1755175</th>\n",
              "      <th>GSM1755176</th>\n",
              "      <th>GSM1755177</th>\n",
              "      <th>GSM1755178</th>\n",
              "      <th>GSM1755179</th>\n",
              "      <th>GSM1755180</th>\n",
              "      <th>GSM1755181</th>\n",
              "      <th>GSM1755182</th>\n",
              "      <th>GSM1755183</th>\n",
              "      <th>GSM1755184</th>\n",
              "      <th>GSM1755185</th>\n",
              "      <th>GSM1755186</th>\n",
              "      <th>GSM1755187</th>\n",
              "      <th>GSM1755188</th>\n",
              "      <th>GSM1755189</th>\n",
              "      <th>GSM1755190</th>\n",
              "      <th>GSM1755191</th>\n",
              "      <th>GSM1755192</th>\n",
              "      <th>GSM1755193</th>\n",
              "      <th>GSM1755194</th>\n",
              "      <th>GSM1755195</th>\n",
              "      <th>GSM1755196</th>\n",
              "      <th>GSM1755197</th>\n",
              "      <th>GSM1755198</th>\n",
              "      <th>GSM1755199</th>\n",
              "      <th>GSM1755200</th>\n",
              "      <th>GSM1755201</th>\n",
              "      <th>GSM1755202</th>\n",
              "      <th>GSM1755203</th>\n",
              "      <th>GSM1755204</th>\n",
              "      <th>GSM1755205</th>\n",
              "      <th>GSM1755206</th>\n",
              "      <th>GSM1755208</th>\n",
              "      <th>GSM1755209</th>\n",
              "      <th>GSM1755210</th>\n",
              "      <th>...</th>\n",
              "      <th>GSM1755367</th>\n",
              "      <th>GSM1755371</th>\n",
              "      <th>GSM1755375</th>\n",
              "      <th>GSM1755378</th>\n",
              "      <th>GSM1755382</th>\n",
              "      <th>GSM1755385</th>\n",
              "      <th>GSM1755388</th>\n",
              "      <th>GSM1755392</th>\n",
              "      <th>GSM1755395</th>\n",
              "      <th>GSM2051672</th>\n",
              "      <th>GSM2051673</th>\n",
              "      <th>GSM2051674</th>\n",
              "      <th>GSM2051675</th>\n",
              "      <th>GSM2051676</th>\n",
              "      <th>GSM2051677</th>\n",
              "      <th>GSM2051678</th>\n",
              "      <th>GSM2051679</th>\n",
              "      <th>GSM2051680</th>\n",
              "      <th>GSM2051681</th>\n",
              "      <th>GSM2051682</th>\n",
              "      <th>GSM2051683</th>\n",
              "      <th>GSM2051684</th>\n",
              "      <th>GSM2051685</th>\n",
              "      <th>GSM2051686</th>\n",
              "      <th>GSM2051687</th>\n",
              "      <th>GSM2051688</th>\n",
              "      <th>GSM2051689</th>\n",
              "      <th>GSM2051690</th>\n",
              "      <th>GSM2051691</th>\n",
              "      <th>GSM2051692</th>\n",
              "      <th>GSM2051693</th>\n",
              "      <th>GSM2051694</th>\n",
              "      <th>GSM2051695</th>\n",
              "      <th>GSM2051696</th>\n",
              "      <th>GSM2051697</th>\n",
              "      <th>GSM2051698</th>\n",
              "      <th>GSM2051699</th>\n",
              "      <th>GSM2051700</th>\n",
              "      <th>GSM2051701</th>\n",
              "      <th>GSM2051702</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ID_REF</th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>cg00000029</th>\n",
              "      <td>0.233639</td>\n",
              "      <td>0.707772</td>\n",
              "      <td>0.251903</td>\n",
              "      <td>0.526724</td>\n",
              "      <td>0.404274</td>\n",
              "      <td>0.356997</td>\n",
              "      <td>0.444767</td>\n",
              "      <td>0.188338</td>\n",
              "      <td>0.689528</td>\n",
              "      <td>0.222346</td>\n",
              "      <td>0.748297</td>\n",
              "      <td>0.494161</td>\n",
              "      <td>0.428071</td>\n",
              "      <td>0.283786</td>\n",
              "      <td>0.743197</td>\n",
              "      <td>0.302302</td>\n",
              "      <td>0.437280</td>\n",
              "      <td>0.315523</td>\n",
              "      <td>0.591644</td>\n",
              "      <td>0.351332</td>\n",
              "      <td>0.408740</td>\n",
              "      <td>0.249594</td>\n",
              "      <td>0.240342</td>\n",
              "      <td>0.170581</td>\n",
              "      <td>0.317655</td>\n",
              "      <td>0.244369</td>\n",
              "      <td>0.573248</td>\n",
              "      <td>0.314415</td>\n",
              "      <td>0.329547</td>\n",
              "      <td>0.652745</td>\n",
              "      <td>0.740240</td>\n",
              "      <td>0.370276</td>\n",
              "      <td>0.391482</td>\n",
              "      <td>0.303231</td>\n",
              "      <td>0.164043</td>\n",
              "      <td>0.114542</td>\n",
              "      <td>0.262227</td>\n",
              "      <td>0.676064</td>\n",
              "      <td>0.326391</td>\n",
              "      <td>0.341752</td>\n",
              "      <td>...</td>\n",
              "      <td>0.470345</td>\n",
              "      <td>0.442620</td>\n",
              "      <td>0.549346</td>\n",
              "      <td>0.389695</td>\n",
              "      <td>0.297349</td>\n",
              "      <td>0.516124</td>\n",
              "      <td>0.384618</td>\n",
              "      <td>0.523748</td>\n",
              "      <td>0.204516</td>\n",
              "      <td>0.577293</td>\n",
              "      <td>0.330351</td>\n",
              "      <td>0.271943</td>\n",
              "      <td>0.390737</td>\n",
              "      <td>0.589534</td>\n",
              "      <td>0.145170</td>\n",
              "      <td>0.265167</td>\n",
              "      <td>0.298249</td>\n",
              "      <td>0.316445</td>\n",
              "      <td>0.340578</td>\n",
              "      <td>0.394026</td>\n",
              "      <td>0.512050</td>\n",
              "      <td>0.276085</td>\n",
              "      <td>0.414950</td>\n",
              "      <td>0.542832</td>\n",
              "      <td>0.162054</td>\n",
              "      <td>0.462029</td>\n",
              "      <td>0.244735</td>\n",
              "      <td>0.332264</td>\n",
              "      <td>0.393462</td>\n",
              "      <td>0.569178</td>\n",
              "      <td>0.209321</td>\n",
              "      <td>0.352109</td>\n",
              "      <td>0.439519</td>\n",
              "      <td>0.256184</td>\n",
              "      <td>0.186200</td>\n",
              "      <td>0.211489</td>\n",
              "      <td>0.148386</td>\n",
              "      <td>0.293506</td>\n",
              "      <td>0.426848</td>\n",
              "      <td>0.375697</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>cg00000108</th>\n",
              "      <td>0.938803</td>\n",
              "      <td>0.911581</td>\n",
              "      <td>0.929206</td>\n",
              "      <td>0.935562</td>\n",
              "      <td>0.941001</td>\n",
              "      <td>0.925437</td>\n",
              "      <td>0.913178</td>\n",
              "      <td>0.917131</td>\n",
              "      <td>0.953745</td>\n",
              "      <td>0.963604</td>\n",
              "      <td>0.930982</td>\n",
              "      <td>0.949970</td>\n",
              "      <td>0.941192</td>\n",
              "      <td>0.908703</td>\n",
              "      <td>0.935924</td>\n",
              "      <td>0.936123</td>\n",
              "      <td>0.961198</td>\n",
              "      <td>0.842026</td>\n",
              "      <td>0.949979</td>\n",
              "      <td>0.893223</td>\n",
              "      <td>0.931049</td>\n",
              "      <td>0.947004</td>\n",
              "      <td>0.907066</td>\n",
              "      <td>0.940768</td>\n",
              "      <td>0.935493</td>\n",
              "      <td>0.929884</td>\n",
              "      <td>0.915023</td>\n",
              "      <td>0.950989</td>\n",
              "      <td>0.955798</td>\n",
              "      <td>0.925292</td>\n",
              "      <td>0.937417</td>\n",
              "      <td>0.936939</td>\n",
              "      <td>0.936586</td>\n",
              "      <td>0.924737</td>\n",
              "      <td>0.883539</td>\n",
              "      <td>0.951436</td>\n",
              "      <td>0.921340</td>\n",
              "      <td>0.915088</td>\n",
              "      <td>0.889984</td>\n",
              "      <td>0.920692</td>\n",
              "      <td>...</td>\n",
              "      <td>0.915920</td>\n",
              "      <td>0.855734</td>\n",
              "      <td>0.835532</td>\n",
              "      <td>0.870266</td>\n",
              "      <td>0.880080</td>\n",
              "      <td>0.851440</td>\n",
              "      <td>0.938475</td>\n",
              "      <td>0.901274</td>\n",
              "      <td>0.940875</td>\n",
              "      <td>0.855938</td>\n",
              "      <td>0.821875</td>\n",
              "      <td>0.891613</td>\n",
              "      <td>0.879165</td>\n",
              "      <td>0.804980</td>\n",
              "      <td>0.855547</td>\n",
              "      <td>0.862085</td>\n",
              "      <td>0.903575</td>\n",
              "      <td>0.870697</td>\n",
              "      <td>0.890521</td>\n",
              "      <td>0.900931</td>\n",
              "      <td>0.886325</td>\n",
              "      <td>0.880761</td>\n",
              "      <td>0.876552</td>\n",
              "      <td>0.852082</td>\n",
              "      <td>0.871461</td>\n",
              "      <td>0.855256</td>\n",
              "      <td>0.782079</td>\n",
              "      <td>0.862102</td>\n",
              "      <td>0.862107</td>\n",
              "      <td>0.832371</td>\n",
              "      <td>0.879776</td>\n",
              "      <td>0.752832</td>\n",
              "      <td>0.839864</td>\n",
              "      <td>0.733557</td>\n",
              "      <td>0.868390</td>\n",
              "      <td>0.787937</td>\n",
              "      <td>0.864105</td>\n",
              "      <td>0.788029</td>\n",
              "      <td>0.828619</td>\n",
              "      <td>0.825274</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>cg00000109</th>\n",
              "      <td>0.883326</td>\n",
              "      <td>0.877838</td>\n",
              "      <td>0.731915</td>\n",
              "      <td>0.752427</td>\n",
              "      <td>0.796499</td>\n",
              "      <td>0.887018</td>\n",
              "      <td>0.838659</td>\n",
              "      <td>0.850539</td>\n",
              "      <td>0.893031</td>\n",
              "      <td>0.906258</td>\n",
              "      <td>0.801748</td>\n",
              "      <td>0.429852</td>\n",
              "      <td>0.833911</td>\n",
              "      <td>0.818169</td>\n",
              "      <td>0.537447</td>\n",
              "      <td>0.870020</td>\n",
              "      <td>0.888209</td>\n",
              "      <td>0.819853</td>\n",
              "      <td>0.631496</td>\n",
              "      <td>0.604464</td>\n",
              "      <td>0.849264</td>\n",
              "      <td>0.841755</td>\n",
              "      <td>0.882378</td>\n",
              "      <td>0.902550</td>\n",
              "      <td>0.881787</td>\n",
              "      <td>0.717837</td>\n",
              "      <td>0.738306</td>\n",
              "      <td>0.868691</td>\n",
              "      <td>0.610745</td>\n",
              "      <td>0.916624</td>\n",
              "      <td>0.628656</td>\n",
              "      <td>0.873302</td>\n",
              "      <td>0.891642</td>\n",
              "      <td>0.853650</td>\n",
              "      <td>0.583921</td>\n",
              "      <td>0.856182</td>\n",
              "      <td>0.912622</td>\n",
              "      <td>0.781826</td>\n",
              "      <td>0.813454</td>\n",
              "      <td>0.899448</td>\n",
              "      <td>...</td>\n",
              "      <td>0.548857</td>\n",
              "      <td>0.808218</td>\n",
              "      <td>0.559361</td>\n",
              "      <td>0.611600</td>\n",
              "      <td>0.641877</td>\n",
              "      <td>0.822538</td>\n",
              "      <td>0.583476</td>\n",
              "      <td>0.787079</td>\n",
              "      <td>0.870045</td>\n",
              "      <td>0.735572</td>\n",
              "      <td>0.641026</td>\n",
              "      <td>0.596404</td>\n",
              "      <td>0.829746</td>\n",
              "      <td>0.535771</td>\n",
              "      <td>0.789474</td>\n",
              "      <td>0.830111</td>\n",
              "      <td>0.824748</td>\n",
              "      <td>0.500782</td>\n",
              "      <td>0.832552</td>\n",
              "      <td>0.875909</td>\n",
              "      <td>0.787627</td>\n",
              "      <td>0.811469</td>\n",
              "      <td>0.815374</td>\n",
              "      <td>0.833258</td>\n",
              "      <td>0.845466</td>\n",
              "      <td>0.826607</td>\n",
              "      <td>0.692812</td>\n",
              "      <td>0.481772</td>\n",
              "      <td>0.824240</td>\n",
              "      <td>0.822971</td>\n",
              "      <td>0.816227</td>\n",
              "      <td>0.590704</td>\n",
              "      <td>0.834401</td>\n",
              "      <td>0.740834</td>\n",
              "      <td>0.682739</td>\n",
              "      <td>0.705098</td>\n",
              "      <td>0.810355</td>\n",
              "      <td>0.598137</td>\n",
              "      <td>0.551450</td>\n",
              "      <td>0.750243</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>cg00000165</th>\n",
              "      <td>0.528943</td>\n",
              "      <td>0.132206</td>\n",
              "      <td>0.670318</td>\n",
              "      <td>0.205442</td>\n",
              "      <td>0.424001</td>\n",
              "      <td>0.151361</td>\n",
              "      <td>0.371850</td>\n",
              "      <td>0.673996</td>\n",
              "      <td>0.858881</td>\n",
              "      <td>0.517007</td>\n",
              "      <td>0.338881</td>\n",
              "      <td>0.530169</td>\n",
              "      <td>0.123305</td>\n",
              "      <td>0.147213</td>\n",
              "      <td>0.880645</td>\n",
              "      <td>0.859784</td>\n",
              "      <td>0.093326</td>\n",
              "      <td>0.110808</td>\n",
              "      <td>0.193867</td>\n",
              "      <td>0.758933</td>\n",
              "      <td>0.276771</td>\n",
              "      <td>0.696539</td>\n",
              "      <td>0.597661</td>\n",
              "      <td>0.562081</td>\n",
              "      <td>0.840610</td>\n",
              "      <td>0.106546</td>\n",
              "      <td>0.111664</td>\n",
              "      <td>0.449088</td>\n",
              "      <td>0.112873</td>\n",
              "      <td>0.543175</td>\n",
              "      <td>0.610725</td>\n",
              "      <td>0.211313</td>\n",
              "      <td>0.816915</td>\n",
              "      <td>0.809718</td>\n",
              "      <td>0.204638</td>\n",
              "      <td>0.938450</td>\n",
              "      <td>0.702455</td>\n",
              "      <td>0.448489</td>\n",
              "      <td>0.096944</td>\n",
              "      <td>0.778999</td>\n",
              "      <td>...</td>\n",
              "      <td>0.450405</td>\n",
              "      <td>0.387707</td>\n",
              "      <td>0.789935</td>\n",
              "      <td>0.567875</td>\n",
              "      <td>0.470227</td>\n",
              "      <td>0.475603</td>\n",
              "      <td>0.166931</td>\n",
              "      <td>0.218099</td>\n",
              "      <td>0.872623</td>\n",
              "      <td>0.190535</td>\n",
              "      <td>0.380332</td>\n",
              "      <td>0.103104</td>\n",
              "      <td>0.090690</td>\n",
              "      <td>0.557864</td>\n",
              "      <td>0.744303</td>\n",
              "      <td>0.470095</td>\n",
              "      <td>0.741788</td>\n",
              "      <td>0.162595</td>\n",
              "      <td>0.441244</td>\n",
              "      <td>0.101852</td>\n",
              "      <td>0.157865</td>\n",
              "      <td>0.624784</td>\n",
              "      <td>0.372658</td>\n",
              "      <td>0.713473</td>\n",
              "      <td>0.771577</td>\n",
              "      <td>0.357595</td>\n",
              "      <td>0.178367</td>\n",
              "      <td>0.183553</td>\n",
              "      <td>0.528264</td>\n",
              "      <td>0.720386</td>\n",
              "      <td>0.098405</td>\n",
              "      <td>0.452315</td>\n",
              "      <td>0.279491</td>\n",
              "      <td>0.103816</td>\n",
              "      <td>0.730732</td>\n",
              "      <td>0.237288</td>\n",
              "      <td>0.588285</td>\n",
              "      <td>0.462184</td>\n",
              "      <td>0.136910</td>\n",
              "      <td>0.552412</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>cg00000236</th>\n",
              "      <td>0.634338</td>\n",
              "      <td>0.744383</td>\n",
              "      <td>0.876071</td>\n",
              "      <td>0.822100</td>\n",
              "      <td>0.903711</td>\n",
              "      <td>0.909017</td>\n",
              "      <td>0.813823</td>\n",
              "      <td>0.651720</td>\n",
              "      <td>0.895219</td>\n",
              "      <td>0.887799</td>\n",
              "      <td>0.908742</td>\n",
              "      <td>0.853340</td>\n",
              "      <td>0.315376</td>\n",
              "      <td>0.844244</td>\n",
              "      <td>0.838710</td>\n",
              "      <td>0.853366</td>\n",
              "      <td>0.877501</td>\n",
              "      <td>0.607351</td>\n",
              "      <td>0.891350</td>\n",
              "      <td>0.829340</td>\n",
              "      <td>0.869451</td>\n",
              "      <td>0.865787</td>\n",
              "      <td>0.853516</td>\n",
              "      <td>0.878366</td>\n",
              "      <td>0.824250</td>\n",
              "      <td>0.365203</td>\n",
              "      <td>0.502188</td>\n",
              "      <td>0.843175</td>\n",
              "      <td>0.787488</td>\n",
              "      <td>0.865691</td>\n",
              "      <td>0.903356</td>\n",
              "      <td>0.826029</td>\n",
              "      <td>0.858634</td>\n",
              "      <td>0.858712</td>\n",
              "      <td>0.875967</td>\n",
              "      <td>0.929785</td>\n",
              "      <td>0.574134</td>\n",
              "      <td>0.864969</td>\n",
              "      <td>0.708163</td>\n",
              "      <td>0.893024</td>\n",
              "      <td>...</td>\n",
              "      <td>0.873256</td>\n",
              "      <td>0.808309</td>\n",
              "      <td>0.812656</td>\n",
              "      <td>0.823338</td>\n",
              "      <td>0.394684</td>\n",
              "      <td>0.771452</td>\n",
              "      <td>0.836030</td>\n",
              "      <td>0.869506</td>\n",
              "      <td>0.848087</td>\n",
              "      <td>0.804929</td>\n",
              "      <td>0.763158</td>\n",
              "      <td>0.781875</td>\n",
              "      <td>0.759177</td>\n",
              "      <td>0.773295</td>\n",
              "      <td>0.687826</td>\n",
              "      <td>0.516738</td>\n",
              "      <td>0.816484</td>\n",
              "      <td>0.842311</td>\n",
              "      <td>0.828511</td>\n",
              "      <td>0.824529</td>\n",
              "      <td>0.802901</td>\n",
              "      <td>0.754381</td>\n",
              "      <td>0.800513</td>\n",
              "      <td>0.811217</td>\n",
              "      <td>0.772063</td>\n",
              "      <td>0.521298</td>\n",
              "      <td>0.752059</td>\n",
              "      <td>0.790965</td>\n",
              "      <td>0.639010</td>\n",
              "      <td>0.538818</td>\n",
              "      <td>0.812312</td>\n",
              "      <td>0.731875</td>\n",
              "      <td>0.801151</td>\n",
              "      <td>0.715255</td>\n",
              "      <td>0.690355</td>\n",
              "      <td>0.616168</td>\n",
              "      <td>0.421314</td>\n",
              "      <td>0.762565</td>\n",
              "      <td>0.760895</td>\n",
              "      <td>0.769858</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5 rows × 150 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "            GSM1755170  GSM1755171  ...  GSM2051701  GSM2051702\n",
              "ID_REF                              ...                        \n",
              "cg00000029    0.233639    0.707772  ...    0.426848    0.375697\n",
              "cg00000108    0.938803    0.911581  ...    0.828619    0.825274\n",
              "cg00000109    0.883326    0.877838  ...    0.551450    0.750243\n",
              "cg00000165    0.528943    0.132206  ...    0.136910    0.552412\n",
              "cg00000236    0.634338    0.744383  ...    0.760895    0.769858\n",
              "\n",
              "[5 rows x 150 columns]"
            ]
          },
          "execution_count": 15,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "meth.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-nS-HMjWRUK7"
      },
      "source": [
        "Thte meta file is loaded in here along with other variable to include in the autogluon training in case the help. "
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 238
        },
        "id": "pC2XoWnSPeNe",
        "outputId": "45f58a18-3449-4c37-e336-3762f0ab7d80"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Molecular Subgroup (Consensus)</th>\n",
              "      <th>Gender (F= FeMale, M=Male)</th>\n",
              "      <th>Age at diagnosis</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>Sample Name</th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>dkfz_ATRT_1</th>\n",
              "      <td>SHH</td>\n",
              "      <td>F</td>\n",
              "      <td>2.0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>dkfz_ATRT_2</th>\n",
              "      <td>SHH</td>\n",
              "      <td>M</td>\n",
              "      <td>2.0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>dkfz_ATRT_3</th>\n",
              "      <td>SHH</td>\n",
              "      <td>F</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>dkfz_ATRT_4</th>\n",
              "      <td>MYC</td>\n",
              "      <td>M</td>\n",
              "      <td>4.0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>dkfz_ATRT_5</th>\n",
              "      <td>SHH</td>\n",
              "      <td>M</td>\n",
              "      <td>4.0</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "            Molecular Subgroup (Consensus)  ... Age at diagnosis\n",
              "Sample Name                                 ...                 \n",
              "dkfz_ATRT_1                            SHH  ...              2.0\n",
              "dkfz_ATRT_2                            SHH  ...              2.0\n",
              "dkfz_ATRT_3                            SHH  ...              NaN\n",
              "dkfz_ATRT_4                            MYC  ...              4.0\n",
              "dkfz_ATRT_5                            SHH  ...              4.0\n",
              "\n",
              "[5 rows x 3 columns]"
            ]
          },
          "execution_count": 16,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "classfile = DATA / \"metaandclassify.txt\"\n",
        "classified = pd.read_csv(classfile, sep=\"\\t\", comment=\"#\",index_col=0)\n",
        "classified = classified[['Molecular Subgroup (Consensus)','Gender (F= FeMale, M=Male)','Age at diagnosis']]\n",
        "classified.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 400
        },
        "id": "zXIv1coKSmgS",
        "outputId": "84d4469d-7a6f-4bc0-d5cd-0dbc02f81fcd"
      },
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th>!Sample_title</th>\n",
              "      <th>SampleGeo</th>\n",
              "      <th>!Sample_status</th>\n",
              "      <th>!Sample_submission_date</th>\n",
              "      <th>!Sample_last_update_date</th>\n",
              "      <th>!Sample_type</th>\n",
              "      <th>!Sample_channel_count</th>\n",
              "      <th>!Sample_source_name_ch1</th>\n",
              "      <th>!Sample_organism_ch1</th>\n",
              "      <th>cha1</th>\n",
              "      <th>cha1</th>\n",
              "      <th>cha1</th>\n",
              "      <th>!Sample_treatment_protocol_ch1</th>\n",
              "      <th>!Sample_growth_protocol_ch1</th>\n",
              "      <th>!Sample_molecule_ch1</th>\n",
              "      <th>!Sample_extract_protocol_ch1</th>\n",
              "      <th>!Sample_label_ch1</th>\n",
              "      <th>!Sample_label_protocol_ch1</th>\n",
              "      <th>!Sample_taxid_ch1</th>\n",
              "      <th>!Sample_hyb_protocol</th>\n",
              "      <th>!Sample_scan_protocol</th>\n",
              "      <th>!Sample_description</th>\n",
              "      <th>!Sample_data_processing</th>\n",
              "      <th>!Sample_platform_id</th>\n",
              "      <th>!Sample_contact_name</th>\n",
              "      <th>!Sample_contact_email</th>\n",
              "      <th>!Sample_contact_department</th>\n",
              "      <th>!Sample_contact_institute</th>\n",
              "      <th>!Sample_contact_address</th>\n",
              "      <th>!Sample_contact_city</th>\n",
              "      <th>!Sample_contact_zip/postal_code</th>\n",
              "      <th>!Sample_contact_country</th>\n",
              "      <th>!Sample_supplementary_file</th>\n",
              "      <th>!Sample_data_row_count</th>\n",
              "      <th>!Sample_relation</th>\n",
              "      <th>!series_matrix_table_begin</th>\n",
              "      <th>ID_REF</th>\n",
              "      <th>Class</th>\n",
              "      <th>Age</th>\n",
              "      <th>Gender</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>dkfz_ATRT_10</th>\n",
              "      <td>GSM1816342</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>RNA</td>\n",
              "      <td>1</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>biotin</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>9606</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>69120</td>\n",
              "      <td>Germany</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>54675</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GSM1816342</td>\n",
              "      <td>TYR</td>\n",
              "      <td>NaN</td>\n",
              "      <td>F</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>dkfz_ATRT_101</th>\n",
              "      <td>GSM1816360</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>RNA</td>\n",
              "      <td>1</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>gender: F</td>\n",
              "      <td>age (years): NA</td>\n",
              "      <td>tumor location: NA</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>biotin</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>9606</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>69120</td>\n",
              "      <td>Germany</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>54675</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GSM1816360</td>\n",
              "      <td>TYR</td>\n",
              "      <td>NaN</td>\n",
              "      <td>F</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>dkfz_ATRT_11</th>\n",
              "      <td>GSM1816346</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>RNA</td>\n",
              "      <td>1</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>age (years): 2</td>\n",
              "      <td>tumor location: Infratentorial</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>biotin</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>9606</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>69120</td>\n",
              "      <td>Germany</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>54675</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GSM1816346</td>\n",
              "      <td>SHH</td>\n",
              "      <td>2.0</td>\n",
              "      <td>M</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>dkfz_ATRT_12</th>\n",
              "      <td>GSM1816347</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>RNA</td>\n",
              "      <td>1</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>gender: M</td>\n",
              "      <td>age (years): 1</td>\n",
              "      <td>tumor location: Infratentorial</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>biotin</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>9606</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>69120</td>\n",
              "      <td>Germany</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>54675</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GSM1816347</td>\n",
              "      <td>SHH</td>\n",
              "      <td>1.0</td>\n",
              "      <td>M</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>dkfz_ATRT_14</th>\n",
              "      <td>GSM1816377</td>\n",
              "      <td>Public on Mar 02 2016</td>\n",
              "      <td>Jul 09 2015</td>\n",
              "      <td>Mar 13 2018</td>\n",
              "      <td>RNA</td>\n",
              "      <td>1</td>\n",
              "      <td>Primary ATRT tumor tissue</td>\n",
              "      <td>Homo sapiens</td>\n",
              "      <td>gender: NA</td>\n",
              "      <td>age (years): 2</td>\n",
              "      <td>tumor location: Infratentorial</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>total RNA</td>\n",
              "      <td>Total RNA from biopsy samples was isolated aft...</td>\n",
              "      <td>biotin</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>9606</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Affymetrix protocol</td>\n",
              "      <td>Gene expression data from an ATRT tumor sample</td>\n",
              "      <td>For normalization of the expression data we us...</td>\n",
              "      <td>GPL570</td>\n",
              "      <td>Marcel,,Kool</td>\n",
              "      <td>m.kool@dkfz.de</td>\n",
              "      <td>Pediatric Neurooncology</td>\n",
              "      <td>German Cancer Research Center DKFZ</td>\n",
              "      <td>Im Neuenheimer Feld 280</td>\n",
              "      <td>Heidelberg</td>\n",
              "      <td>69120</td>\n",
              "      <td>Germany</td>\n",
              "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1816...</td>\n",
              "      <td>54675</td>\n",
              "      <td>Reanalyzed by: GSE111787</td>\n",
              "      <td>NaN</td>\n",
              "      <td>GSM1816377</td>\n",
              "      <td>TYR</td>\n",
              "      <td>NaN</td>\n",
              "      <td>M</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "!Sample_title   SampleGeo         !Sample_status  ...  Age Gender\n",
              "dkfz_ATRT_10   GSM1816342  Public on Mar 02 2016  ...  NaN      F\n",
              "dkfz_ATRT_101  GSM1816360  Public on Mar 02 2016  ...  NaN      F\n",
              "dkfz_ATRT_11   GSM1816346  Public on Mar 02 2016  ...  2.0      M\n",
              "dkfz_ATRT_12   GSM1816347  Public on Mar 02 2016  ...  1.0      M\n",
              "dkfz_ATRT_14   GSM1816377  Public on Mar 02 2016  ...  NaN      M\n",
              "\n",
              "[5 rows x 39 columns]"
            ]
          },
          "execution_count": 17,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "meta = gemeta.transpose()\n",
        "classified = classified[classified.index.isin(meta.index)]\n",
        "classified.sort_index(inplace=True)\n",
        "meta.sort_index(inplace=True)\n",
        "meta['Class'] = classified['Molecular Subgroup (Consensus)']\n",
        "meta['Age'] = classified['Age at diagnosis']\n",
        "meta['Gender'] = classified['Gender (F= FeMale, M=Male)']\n",
        "meta = meta.rename(columns={\"!Sample_geo_accession\": \"SampleGeo\"})\n",
        "meta.rename(columns = {list(meta)[9]: 'cha1'}, inplace = True)\n",
        "\n",
        "meta.head()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "mtRVGnInZx-q"
      },
      "source": [
        "### Split test set"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "rk670ue7Z0YZ"
      },
      "source": [
        "3 samples are chosen from each class of tumor to become the test dataset"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "KQFWbwa-i3Yy",
        "outputId": "47c957af-7f05-4936-80b6-65b9300edc96"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "Class              \n",
              "MYC    dkfz_ATRT_46    GSM1816359\n",
              "       dkfz_ATRT_49    GSM1816356\n",
              "       dkfz_ATRT_90    GSM1816380\n",
              "SHH    dkfz_ATRT_51    GSM1816357\n",
              "       dkfz_ATRT_7     GSM1816354\n",
              "       dkfz_ATRT_72    GSM1816372\n",
              "TYR    dkfz_ATRT_85    GSM1816374\n",
              "       dkfz_ATRT_86    GSM1816341\n",
              "       dkfz_ATRT_88    GSM1816371\n",
              "Name: SampleGeo, dtype: object"
            ]
          },
          "execution_count": 18,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "\n",
        "# pick 3 samples from each class (MYC, SHH, TYR) as test set for evaluation in the end\n",
        "test_sample_id = meta.groupby(\"Class\").apply(lambda g: g.SampleGeo.iloc[-3:])\n",
        "test_sample_id"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "Ui84gx5sZBdh"
      },
      "outputs": [],
      "source": [
        "# define train and test set\n",
        "meta[\"subset\"] = \"train\"\n",
        "meta.loc[meta.SampleGeo.isin(test_sample_id), \"subset\"] = \"test\""
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "vBpq7fQAbbR_",
        "outputId": "9a59ea5c-ef62-4e1c-b768-bdbc0a0ca5e1"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "subset  Class\n",
              "test    MYC       3\n",
              "        SHH       3\n",
              "        TYR       3\n",
              "train   MYC      12\n",
              "        SHH      13\n",
              "        TYR      15\n",
              "Name: SampleGeo, dtype: int64"
            ]
          },
          "execution_count": 20,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "meta.groupby([\"subset\", \"Class\"]).SampleGeo.nunique()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "RRsfR-hnfBTI"
      },
      "source": [
        "### Feature selection"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "1LjjwB2S__xE"
      },
      "source": [
        "Using the method from class we calculate the mutual information between each gene and the severity using scikit-learn's [`mutual_info_classif`](https://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.mutual_info_classif.html#sklearn.feature_selection.mutual_info_classif) function. All genes are ranked and a cutoff is selected. "
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "7yvHoBURln2b"
      },
      "outputs": [],
      "source": [
        "#=======================================================\n",
        "# Your code here\n",
        "# Perform feature selection using mutual_info_classif \n",
        "#=======================================================\n",
        "di = {\"TYR\": 0, \"SHH\": 1,\"MYC\":2}\n",
        "meta = meta.replace({\"Class\": di})\n",
        "target = (meta[\"Class\"]).astype(int)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "SB_8-_N-hjtU"
      },
      "outputs": [],
      "source": [
        "# Creating trianing and testing data. \n",
        "\n",
        "\n",
        "y_train = target[(meta.subset == \"train\")].values\n",
        "y_test = target[(meta.subset == \"test\")].values\n",
        "x_train = ge[meta[meta.subset == \"train\"].SampleGeo].transpose()\n",
        "x_test = ge[meta[meta.subset == \"test\"].SampleGeo].transpose()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "u-7i-t_yQ2Nr"
      },
      "outputs": [],
      "source": [
        "from sklearn.feature_selection import mutual_info_classif\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "MMpI6pVMQ6aE"
      },
      "outputs": [],
      "source": [
        "midf = pd.DataFrame({\n",
        "    \"feature\": x_train.columns,\n",
        "    \"mi\": mutual_info_classif(x_train, y_train,)\n",
        "}).sort_values(\"mi\", ascending=False)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "Lx-P5mAPYUhx",
        "outputId": "8a06904a-3c36-40f7-8265-ec18aae9b4f8"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "[<matplotlib.lines.Line2D at 0x7f93d11da3d0>]"
            ]
          },
          "execution_count": null,
          "metadata": {},
          "output_type": "execute_result"
        },
        {
          "data": {
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXoAAAD4CAYAAADiry33AAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAgAElEQVR4nO3deXxV9Z3/8dfnbkludrIQlgBhB0EWI1VxXypqf+BMpyrOtHZq67QzttaZdn522s6vYzszdtpObaszrWPp2BWt2hardQUXXCihsoclsgYCCUsWyJ77/f1xLxgigUgSTnLu+/l45JGc5Saf8zjwvt/7Pd/zPeacQ0RE/CvgdQEiItK/FPQiIj6noBcR8TkFvYiIzynoRUR8LuR1AV3l5+e7MWPGeF2GiMigsmrVqgPOuYKTbRtwQT9mzBjKysq8LkNEZFAxs53dbVPXjYiIzynoRUR8TkEvIuJzCnoREZ9T0IuI+JyCXkTE5xT0IiI+55ugr29u4/4Xt7Bmd63XpYiIDCi+CXrn4P4Xt7JyxyGvSxERGVB8E/RZqSEioQA1DS1elyIiMqD4JujNjMLMFKoV9CIiJ/BN0AMUZKaoRS8i0oWvgr5QQS8i8h6+CvqCzBSqG5q9LkNEZEDxV9BnpHK4sY3W9pjXpYiIDBi+CvrCrBQADh5V942IyDG+CvqCjHjQV9cr6EVEjvFX0GfGg14XZEVE3uWroD/WdVNzREEvInKMr4I+L11dNyIiXfkq6COhALnRMDVHNMRSROSYHgW9mc0zs81mVmFm95xk+3fNbHXia4uZ1Xba1tFp25K+LP5kCjNT1aIXEekkdLodzCwIPAhcA1QCK81siXNu47F9nHN3d9r/s8CsTr+iyTk3s+9KPrWCzBT10YuIdNKTFv0coMI5t8051wosBhacYv+FwK/6orgzofluRERO1JOgHwHs7rRcmVj3HmY2GigBlnZanWpmZWb2lpnd2M3r7kjsU1ZTU9PD0k/u2AyWzrle/R4REb/o64uxtwCPO+c6Oq0b7ZwrBW4F7jezcV1f5Jx7yDlX6pwrLSgo6FUBBZkptLbHqG9u79XvERHxi54E/R6guNPyyMS6k7mFLt02zrk9ie/bgJc5sf++z+mmKRGRE/Uk6FcCE8ysxMwixMP8PaNnzGwykAu82WldrpmlJH7OB+YCG7u+ti8dC3rNYikiEnfaUTfOuXYzuxN4DggCi5xzG8zsXqDMOXcs9G8BFrsTO8enAD8ysxjxN5X7Oo/W6Q+FatGLiJzgtEEP4Jx7Bnimy7p/7rL8tZO87g1gei/qe98KMlMBBb2IyDG+ujMW9JBwEZGufBf0ZkZBhsbSi4gc47ugh/gsltUKehERwKdBrxa9iMi7/Bn0mu9GROQ4XwZ9YWYqh4626iHhIiL4NOiP3TSlh4SLiPg06HXTlIjIu3wZ9MenQdADSERE/B30uiArIuLToM/PSCEcNJZXHPC6FBERz/ky6COhAHdeMYGn11bxm7crvS5HRMRTvgx6gL+7Yhznj8nlq7/dwK6DjV6XIyLiGd8GfSgY4Ls3z8QMPrf4bdo6NKZeRJKTb4MeYGRulH//8+ms3l3Lw69t97ocERFP+DroAT507nAuHJvHY2W79cBwEUlKvg96gBtnDWf7gaOs21PndSkiImddUgT9vHOGEQkG+N3qvV6XIiJy1iVF0GdHw1w+qYCn1uylI6buGxFJLkkR9AALZo6guqGFFdsOel2KiMhZlTRBf9WUQtIjQXXfiEjSSZqgTw0HuXZaEc+sr6KlvcPrckREzpqkCXqA+TOG09Dczsuba7wuRUTkrEmqoL94fD556RGWrFH3jYgkj6QK+lAwwLXTili2qZrmNnXfiEhySKqgB7huWhGNrR28skXdNyKSHJIu6C8Ym0dONMyz6/d5XYqIyFmRdEEfDga4ZspQXty4X6NvRCQpJF3QA1w/fRgNLe28UaGbp0TE/5Iy6C8an0dmaohn1lV5XYqISL9LyqBPCQW5espQXijfrweSiIjvJWXQQ3z0TW1jG29p7hsR8bmkDfpLJxYQjQT55YpdxDSjpYj4WNIGfWo4yCcvLuEP6/dx16OrNQJHRHwr5HUBXrr7momkRUJ889lN1DQ086OPlpKdFva6LBGRPpW0LXoAM+Mzl4/j/ptnsmrnYf7+0dVelyQi0ueSukV/zI2zRrC2so5frNhJa3uMSCip3/9ExGd6lGhmNs/MNptZhZndc5Lt3zWz1YmvLWZW22nbbWa2NfF1W18W35fOH5NLS3uMDXv1AHER8ZfTtujNLAg8CFwDVAIrzWyJc27jsX2cc3d32v+zwKzEz0OA/weUAg5YlXjt4T49ij5w3uhcAFbtPMysUbkeVyMi0nd60qKfA1Q457Y551qBxcCCU+y/EPhV4udrgRecc4cS4f4CMK83BfeXwqxUioeksWrngHsPEhHplZ4E/Qhgd6flysS69zCz0UAJsPT9vnYgKB09hLKdh3FO4+pFxD/6+qrjLcDjzrn3NSjdzO4wszIzK6up8W6e+Nmjc6lpaKHycJNnNYiI9LWeBP0eoLjT8sjEupO5hXe7bXr8WufcQ865UudcaUFBQQ9K6h/njXq3n15ExC96EvQrgQlmVmJmEeJhvqTrTmY2GcgF3uy0+jngg2aWa2a5wAcT6wakSUWZZKSEKNt5yOtSRET6zGlH3Tjn2s3sTuIBHQQWOec2mNm9QJlz7ljo3wIsdp06uJ1zh8zs68TfLADudc4N2BQNBoxZo3JYtbP29DuLiAwSPbphyjn3DPBMl3X/3GX5a928dhGw6AzrO+vOG53L91/aSkNzG5mpmg5BRAY/3QLaxXmjc4k5WL1brXoR8QcFfRczi3MImC7Iioh/KOi7yEwNM6koS0EvIr6hoD+JD5QMYeWOQzS3aY56ERn8FPQnccXkQprbYrz5jh4zKCKDn4L+JC4YO4RoJMiL5fu9LkVEpNcU9CeREgpyyYR8lm6q1rw3IjLoKei7cdWUoVTVNbNhb73XpYiI9IqCvhtXTCrEDJZuqva6FBGRXlHQd6MgM4UZI3N4Sf30IjLIKehP4eophayprKO6vtnrUkREzpiC/hSumjIUUPeNiAxuCvpTmFyUyfDsVF4sV9CLyOCloD8FM2PetGEs3bSfh1/bpqGWIjIo9Wia4mT2hWsnUlXXxDeeLmfr/iN8/cZpREJ6fxSRwUOJdRrRSIgHb53N564cz6Nlu/noj1fQ2NrudVkiIj2moO+BQMD4+w9O4v6bZ7JyxyHu/OXbtHfEvC5LRKRHFPTvw42zRnDvgmks3VTNP/1mnfrsRWRQUB/9+/RXF4ymur6Z7y+toDAzlS9cO8nrkkRETklBfwbuvmYi++tbeGBZBaPyotxUWux1SSIi3VLXzRkwM77xZ9OYOz6PL/9mHX/cfsjrkkREuqWgP0PhYID/uvU8inOj/M3Pyth1sNHrkkRETkpB3wvZ0TA//vj5xBzc/shKmlr16EERGXgU9L1Ukp/O/TfPZGv1EZ74U6XX5YiIvIeCvg9cPqmAc0dms2j5dmIxDbkUkYFFQd8HzIzbLy5h24GjvLxFE6CJyMCioO8j108fxrDsVB5+bbvXpYiInEBB30fCwQC3XTSGN945yIa9dV6XIyJynIK+Dy08fxTRSJBFy3d4XYqIyHEK+j6UHQ1zU2kxS9bs0eMHRWTAUND3sb+eO4b2mOPnb+30uhQREUBB3+dG56Vz1eRCfrFiF81tuoFKRLynoO8Hn5hbwsGjrSxZs9frUkREFPT94cJxeUwamslPXt+hOetFxHMK+n5gZnzi4jGUV9Xz1jbNbCki3lLQ95MFM0eQGw3zk9d1A5WIeEtB309Sw0H+8gOjeaF8PxXVDV6XIyJJTEHfjz524Wiy08Lc/kgZ1Q0aVy8i3uhR0JvZPDPbbGYVZnZPN/vcZGYbzWyDmf2y0/oOM1ud+FrSV4UPBoVZqSz6+PlU17dw26KV1De3eV2SiCQhO92oEDMLAluAa4BKYCWw0Dm3sdM+E4DHgCudc4fNrNA5V53YdsQ5l9HTgkpLS11ZWdn7P5IB7JUtNXzykZXMGpXL3VdPJCcaJicaJhoJEY0ECQf1wUpEesfMVjnnSk+2rScPB58DVDjntiV+2WJgAbCx0z6fAh50zh0GOBbyEnfZxAK+c9NM7lr8Ngv/5633bE8JBfiX+edwy5xRHlQnIn7Xk6AfAezutFwJfKDLPhMBzOx1IAh8zTn3bGJbqpmVAe3Afc6533b9A2Z2B3AHwKhR/gy7+TOGM6s4h92HGqltaqO2sY3G1nYaWzt4sXw///6HTdxw7jAyU8NelyoiPtOToO/p75kAXA6MBF41s+nOuVpgtHNuj5mNBZaa2Trn3DudX+ycewh4COJdN31U04BTPCRK8ZDoe9ZfMamQ//PAcv739R189qoJHlQmIn7Wk87hPUBxp+WRiXWdVQJLnHNtzrntxPv0JwA45/Ykvm8DXgZm9bJm35k+Mpurpwzlf17bRl2TLtiKSN/qSdCvBCaYWYmZRYBbgK6jZ35LvDWPmeUT78rZZma5ZpbSaf1cTuzbl4TPXz2B+uZ2Fi3XDVYi0rdOG/TOuXbgTuA5oBx4zDm3wczuNbP5id2eAw6a2UZgGfBF59xBYApQZmZrEuvv6zxaR941bUQ2154zlEXLt1PXqFa9iPSd0w6vPNv8OLyyp8qr6rnue6/xsQtHc++CaV6XIyKDyKmGV2oA9wAyZVgWfz13DD99cyc/fOWd079ARKQH+mrUjfSRr94wlQNHWrnvD5vIjYa5+Xx/DjcVkbNHQT/ABALGdz4yg7qmNr705DrKqxrIjUZIDQeYPTqX88cM8bpEERlkFPQDUCQU4Id/NZu/+dkqfrFiJ20d715Hual0JF++firZUd1YJSI9o6AfoKKRED+7PX4DcntHjKMtHfzw1Xd46NVtLNtcwyfmlpCVFiI1FOScEVlMLsryuGIRGag06maQWb+njnueXMv6PfXH10VCAX73d3OZMkxhL5KsTjXqRkE/CDnnqGtqo6U9xuHGVj764z+Skxbmqc9eTGo46HV5IuIBDa/0GTMjJxphaFYqk4uy+M5HZrC1+gjfeFr3oonIeynofeDSiQV86pISfv7WLp7fsM/rckRkgFHQ+8QXr53MtBFZ3Pmrt/n84rd5veIAsdjA6pYTEW9o1I1PREIBHv7Y+Ty4rILfrt7Db1fvJT8jwpi8dIbnpDGuIIO/vngMWZrvXiTp6GKsDzW3dfDchn28sqWGvbVNVNU1s+tQI0VZqfzbn0/nikmFXpcoIn1Mo26ENbtr+cKv17C1+gg3lY7kGzdOJxJSz52IX2jUjTCjOIfff+5iPnP5OB4rq+THmvdeJGko6JNISijI/503mWumDuUHS7eyv77Z65JE5CxQ0Cehr9wwhfYOx31/2OR1KSJyFijok9DovHQ+dWkJv3l7D6t2HvK6HBHpZwr6JPW3l49naFYKX1uykZb2Dq/LEZF+pHH0SSo9JcQ/XT+FuxavpvTrL3LllEKunjKUvIwIEO/Pn1WcQyBgHlcqIr2loE9iC2aOYEh6hN+vqeKF8v38bvXeE7bffnEJX/3QVI+qE5G+oqBPcpdMKOCSCQX8a0eMDXvraWqLd+M8unI3P3l9O382awTTRmR7XKWI9IaCXgAIBQPMKM45vjylKIvXttbw5d+s48m/nUtQXTgig5YuxspJZUfDfOWGqayprOOXf9zldTki0gsKeunWgpnDmTs+j/94dpNurhIZxBT00i0z4+sLptHSHuPq77zCfX/YRHWDAl9ksNGkZnJa5VX1PLisgmfWVREKBpgyLIvcaJictDCXTizgxpkjNAxTxGOavVL6xI4DR/nfN3aw7cBRahtbqa5vYV99M7NH5XDvgmkanSPiIQW99ItYzPHEnyr55rObOHi0ldLRueREI2SnhRmZm8ackiHMHpWrB5aLnAWnCnoNr5QzFggYHykt5oPnFPFfyyp4e3ctuw81sqGpjar6ZpyDSDDA+MIMctPD5KRFmDUqh9svLsFMXT0iZ4uCXnotOy3Ml66fcsK6uqY2Vu08xIpth6ioPkJtUxvrD9fx9LoqAmZ84uISj6oVST4KeukX2Wlhrpw8lCsnDz2+LhZzfPrnq/jXZ8qZOjyLC8bmeVihSPLQ8Eo5awIB4zs3zWD0kCh3/vJP7KvTUE2Rs0EtejmrMlPD/Oij53Hjg69z04/eZHxhBgEzcqJhFs4ZxXmjc70uUcR31KKXs27C0Ewe+MvZ5GdEqG5oZk9tE89v2MeH//sNbvrhm7ywcT+t7TGvyxTxDQ2vlAHhaEs7j67czcOvbWNvXTPpkSBzx+dz+aRCZo3KYUJhBqGg2iUi3dE4ehk02jpivLK5hmWbq1m2qZq9iX78lFCAyUWZ5GekkJEaIjcaYeGcUUwqyvS4YpGBQUEvg5Jzju0HjrJuTx3rKuso31fP4aNtNLS0UdPQQiwGn79mAndcMlatfUl6vb5hyszmAd8DgsDDzrn7TrLPTcDXAAescc7dmlh/G/CVxG7fcM498r6PQJKSmTG2IIOxBRksmDnihG0Hj7Tw1d+t5z+e3czzG/bzwK2zGJkb9ahSkYHttC16MwsCW4BrgEpgJbDQObex0z4TgMeAK51zh82s0DlXbWZDgDKglPgbwCrgPOfc4e7+nlr00lPOOZ5aW8WXn1zHmPx0nvjMRURCatlLcjpVi74n/yvmABXOuW3OuVZgMbCgyz6fAh48FuDOuerE+muBF5xzhxLbXgDmnclBiHRlZsyfMZxvfWQG6/bU8e3nN3tdksiA1JOgHwHs7rRcmVjX2URgopm9bmZvJbp6evpazOwOMyszs7KampqeVy8CzJtWxF9dMIqHXt3GK1v070ekq776nBsCJgCXAwuB/zGznFO+ohPn3EPOuVLnXGlBQUEflSTJ5Cs3TGXi0Az+4bHV7Dx41OtyRAaUnlyM3QMUd1oemVjXWSWwwjnXBmw3sy3Eg38P8fDv/NqXz7RYke6khoP8YOFs5j+wnMu+9TIl+elcMHYIJfnpRCMh0lOCpIWDpISCpIQDpEdCZKWFyUoNkRON6OHn4ms9uRgbIn4x9iriwb0SuNU5t6HTPvOIX6C9zczygbeBmbx7AXZ2Ytc/Eb8Ye6i7v6eLsdIbOw4c5cXy/by17SArth+iobn9tK/JSg1x2aRCrp5SyIXj8shPT9ETs2TQ6dXwSudcu5ndCTxHfHjlIufcBjO7Fyhzzi1JbPugmW0EOoAvOucOJv7414m/OQDce6qQF+mtMfnpfPKSsXzykrHEYo6mtg6OtrZztKWDptYOWto7aG6L0djaTn1zG3WNbWzYW8/STdU8tWYvAKGAkZ+RQk40TDgYIBAw8tIj3HbRGC6dkK+59GXQ0Q1TIkBHzLF6dy3rKmupOdJCdX0LtU1tdMQcHTHH5n0N7KtvZkZxDndeMZ4rJhXoJi0ZUPSEKZHTCAaM80bndjt7Zkt7B0+s2sODyyr41E/LyI2G+eDUIuZNL+LCsXl6XKIMaGrRi7wPbR0xXiqv5g/rq3ipvJojLe1EQgFKR+dy0bg8RuZGycuIkJ+RwvjCDMJq9ctZorluRPpBc1sHb247yPKtB3i94gCb9jWcsD0jJcQFY/O4ZEI+100rojAr1aNKJRko6EXOgrqm+GRrB4+0sK++mbe2HWJ5RQ27DzURDBhXTS7kljnFXDaxUMM5pc+pj17kLMhOC5OdFmZ8YQbA8YnYKqqP8OtVu3liVSXPb9zP8OxUbj5/FDefX0xRtlr50v/Uohc5S1rbY7xUvp9f/nEXr209QMBgclEWYwvSGVeQQV5GhHAwQDgYYHRelNmjctXylx5Ti15kAIiEAlw3fRjXTR/GzoNHeXxVJev21LGmspan11XRtc01JD3CFZMK+UDJEIqyUynKTqUwM4XM1LDeAOR9UYteZABobuvgSEs7bR0xWttjrNtTx4sb97N0UzX1Xe7uNYt3E00uyuSbHz6X0XnpHlUtA4kuxooMUu0dMfbWNrOvvpmquiYOHGmlrrGVw41tLFmzF+cc/3nTTK6eOtTrUsVjCnoRH9p9qJFP/3wVG/bW8+nLxnHXVRNIi+jGrWTV2wePiMgAVDwkyhOfuYhbzi/mh6+8wxXffpnHynbTERtYjTfxnoJeZBBLDQe578Pn8ugdFzA0O5V/fHwt8+5/lZ++uYO6xjavy5MBQl03Ij7hnOOZdft4cFkFG6vqSQkFuGpKIcW5UXKiEfIyInygZIgu3vqUhleKJAEz44Zzh3HDucNYv6eOR1fu5qXy/bxYXk1re+z4fmPyolw+qZBrpg5lTskQzceTBNSiF0kCTa0d7K1rYvnWA7y8uZo3tx2kuS1GdlqYq6YUMndcPqVjchk1JKr59gcpjboRkRM0tXbwypYant+wj5c2VVPXFO/Pz89I4cJxeVw6IZ/LJhZoIrZBRF03InKCtEiQedOKmDetiFjMsaW6gbIdhynbcYjlFQePP20rLRwkGgmSFgmSnnj2bnpKiPGFGVx7ThHnjxmiu3QHAbXoReQEsZijfF89r1ccoKahhcbW+GMYjz2SsaGlnfKqelrbYwxJj7BwTjGfv3qi+vo9pha9iPRYIGCcMzybc4Znd7vPkZZ2Xtlcw1Nr9vLgsndYse0QD9w6W7NxDlBq0YtIryxZs5d7nlhLWjjIF66dRH5GCtFIkNRwkJRQgEgoQEooQFokSDQSIhoOElB3T59Ti15E+s38GcOZOiyTT//8T3zpyXU9ek00EfpZqSGKslMZnpPGqCFRrp8+7Ph8/tJ31KIXkT7R1hFj58GjNLZ20NjaQXNbB63tMVo7YrS0xWhs66CxpZ2jre9+r29qo6quib21zexvaMY5mFMyhJtKixmZm0Zq4mJwQUYKOdGwhn6eglr0ItLvwsEA4wszz/j1NQ0tPL6qksUrd/GFX695z/a0cJBhOakMz047/ilgwczhjCvQJ4DTUYteRAaUWMyxsaqe+qY2mtvjnw7217ewt7aJvbVNVNXFp2yubmghNRTk3gXn8BfnjUz61r5a9CIyaAQCxrQR3Y/4OWZ/fTN3LX6bLz6+ljfeOcgnLykhkngUYzQSJCstTGpY0zaDWvQiMoh1xBw/WLqV77+0lZPNzhwJBchPjzAiN42RudHj/f9+vMlLUyCIiK9t3tfA9gNHaeuI0dYRO36ht76pjZqGFiprm9h9qJGqumbOHZnNv/3Z9B59ahhM1HUjIr42qSiTSUWnvhDsnOOptVXc+9RG5j+wnI9dOIa7r5lIdlr4LFXpHd2zLCJJwcyYP2M4L/3DZdz6gVE88uYOrvj2y/zqj7t8/1QuBb2IJJXstDDfuHE6T915MeMK0vnSk+v40A+W8+z6KmI+DXz10YtI0jrWnfOfz29mx8FGJg7N4BNzS5gyLIvReVGy0wbPTVq6GCsicgrtHTGeXlfFA0sr2Fp95Pj6/IwIN0wfxl+cV8y0EVkDOvQV9CIiPRCLObZWH2HnwaPsOtTI27treWHjflrbY5TkpzMyN43stDBZaWGCidAPBozxhRnMLM5hUlGmZ9M1a9SNiEgPBAL2nhE8dU1tPL22iqWb9nPwaCt7DjdR39x2fNx+a3uMIy3tAKSEAlw5uZA/nz2SyycVDJg5+tWiFxHpBecclYebWL27lpU7DvH02ioOHm1lSHqEC8flce6IbM4dmUNBZoRQIEAwYAzNSiUS6ts3AXXdiIicJW0dMV7dUsOSNXtZtfMwlYeb3rNPWjhI6ZhcLhqXz6xROYwvzCAvPdKrawAKehERjxw80sL6vfXUNbXRnrhzt7yqgTfeOcCW/e9e+M1OC3PpxAJ+sHDWGf0d9dGLiHgkLyOFyyYWnHRbTUML5VX1VFQf4Z2aI/12l26Pgt7M5gHfA4LAw865+7ps/zjwLWBPYtUDzrmHE9s6gGOPndnlnJvfB3WLiAx6BZkpFGQWcGk3bwR95bRBb2ZB4EHgGqASWGlmS5xzG7vs+qhz7s6T/Iom59zM3pcqIiJnoieXfecAFc65bc65VmAxsKB/yxIRkb7Sk6AfAezutFyZWNfVh81srZk9bmbFndanmlmZmb1lZjee7A+Y2R2Jfcpqamp6Xr2IiJxWXw3kfAoY45w7F3gBeKTTttGJK8G3Aveb2biuL3bOPeScK3XOlRYU9G9flYhIsulJ0O8BOrfQR/LuRVcAnHMHnXMticWHgfM6bduT+L4NeBk4s7FDIiJyRnoS9CuBCWZWYmYR4BZgSecdzGxYp8X5QHlifa6ZpSR+zgfmAl0v4oqISD867agb51y7md0JPEd8eOUi59wGM7sXKHPOLQE+Z2bzgXbgEPDxxMunAD8ysxjxN5X7TjJaR0RE+pHujBUR8YFBNQWCmdUAO3vxK/KBA31UzmCRjMcMyXncyXjMkJzH/X6PebRz7qSjWQZc0PeWmZV1967mV8l4zJCcx52MxwzJedx9ecwDY7JkERHpNwp6ERGf82PQP+R1AR5IxmOG5DzuZDxmSM7j7rNj9l0fvYiInMiPLXoREelEQS8i4nO+CXozm2dmm82swszu8bqe/mJmxWa2zMw2mtkGM7srsX6Imb1gZlsT33O9rrWvmVnQzN42s98nlkvMbEXinD+amKLDV8wsJzEj7CYzKzezC/1+rs3s7sS/7fVm9iszS/XjuTazRWZWbWbrO6076bm1uO8njn+tmc1+P3/LF0Hf6eEo1wFTgYVmNtXbqvpNO/APzrmpwAXA3yWO9R7gJefcBOClxLLf3EViHqWEbwLfdc6NBw4Dt3tSVf/6HvCsc24yMIP48fv2XJvZCOBzQKlzbhrxaVduwZ/n+n+BeV3WdXdurwMmJL7uAP77/fwhXwQ9SfRwFOdclXPuT4mfG4j/xx9B/HiPTQ/9CHDSuf8HKzMbCdxAfHZUzMyAK4HHE7v48ZizgUuBHwM451qdc7X4/FwTn4MrzcxCQBSowofn2jn3KvG5wTrr7twuAH7q4t4CcrpMJnlKfgn6nj4cxVfMbAzxaZ9XAEOdc1WJTfuAoR6V1V/uB/4RiCWW84Ba51x7YnCvEkUAAAHdSURBVNmP57wEqAF+kuiyetjM0vHxuU5Ma/5tYBfxgK8DVuH/c31Md+e2Vxnnl6BPOmaWATwBfN45V995m4uPmfXNuFkz+xBQ7Zxb5XUtZ1kImA38t3NuFnCULt00PjzXucRbryXAcCCd93ZvJIW+PLd+CfrTPhzFT8wsTDzkf+GcezKxev+xj3KJ79Ve1dcP5gLzzWwH8W65K4n3XeckPt6DP895JVDpnFuRWH6cePD7+VxfDWx3ztU459qAJ4mff7+f62O6O7e9yji/BP1pH47iF4m+6R8D5c65/+y0aQlwW+Ln24Dfne3a+otz7kvOuZHOuTHEz+1S59xfAsuAv0js5qtjBnDO7QN2m9mkxKqriD+4x7fnmniXzQVmFk38Wz92zL4+1510d26XAB9LjL65AKjr1MVzes45X3wB1wNbgHeAL3tdTz8e58XEP86tBVYnvq4n3mf9ErAVeBEY4nWt/XT8lwO/T/w8FvgjUAH8Gkjxur5+ON6ZQFnifP8WyPX7uQb+BdgErAd+BqT48VwDvyJ+HaKN+Ke327s7t4ARH1n4DrCO+KikHv8tTYEgIuJzfum6ERGRbijoRUR8TkEvIuJzCnoREZ9T0IuI+JyCXkTE5xT0IiI+9/8BNzpkbnU8hE4AAAAASUVORK5CYII=\n",
            "text/plain": [
              "<Figure size 432x288 with 1 Axes>"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ],
      "source": [
        "import matplotlib.pyplot as plt\n",
        "import numpy as np\n",
        "plt.plot(np.arange(100), midf.mi.iloc[:100])"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "MAn9Bc4uY0jb",
        "outputId": "d6389efe-7a91-47eb-c68d-2ad5c9e59b65"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "28"
            ]
          },
          "execution_count": null,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "selected_genes = midf.feature.loc[midf.mi > 0.62]\n",
        "len(selected_genes)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "jN_hp3HwC3Hv"
      },
      "source": [
        "### Training an ATRT classifier using AutoML"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-qeSQp0sC_tj"
      },
      "source": [
        "Procedures used in class were repeated here to create an AutoML model for classifying ATRT data."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "XHZoPJrADGoA",
        "outputId": "6b5b451a-2738-4c7f-9f96-f41f8ff094f0"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "Warning: path already exists! This predictor may overwrite an existing predictor! path=\"good_quality_atrt\"\n",
            "Presets specified: ['good_quality_faster_inference_only_refit']\n",
            "Beginning AutoGluon training ...\n",
            "AutoGluon will save models to \"good_quality_atrt/\"\n",
            "AutoGluon Version:  0.3.1\n",
            "Train Data Rows:    40\n",
            "Train Data Columns: 28\n",
            "Preprocessing data ...\n",
            "AutoGluon infers your prediction problem is: 'multiclass' (because dtype of label-column == int, but few unique label-values observed).\n",
            "\t3 unique label values:  [0, 1, 2]\n",
            "\tIf 'multiclass' is not the correct problem_type, please manually specify the problem_type argument in fit() (You may specify problem_type as one of: ['binary', 'multiclass', 'regression'])\n",
            "Train Data Class Count: 3\n",
            "Using Feature Generators to preprocess the data ...\n",
            "Fitting AutoMLPipelineFeatureGenerator...\n",
            "\tAvailable Memory:                    11895.46 MB\n",
            "\tTrain Data (Original)  Memory Usage: 0.04 MB (0.0% of available memory)\n",
            "\tInferring data type of each feature based on column values. Set feature_metadata_in to manually specify special dtypes of the features.\n",
            "\tStage 1 Generators:\n",
            "\t\tFitting AsTypeFeatureGenerator...\n",
            "\tStage 2 Generators:\n",
            "\t\tFitting FillNaFeatureGenerator...\n",
            "\tStage 3 Generators:\n",
            "\t\tFitting CategoryFeatureGenerator...\n",
            "\t\t\tFitting CategoryMemoryMinimizeFeatureGenerator...\n",
            "\tStage 4 Generators:\n",
            "\t\tFitting DropUniqueFeatureGenerator...\n",
            "\tUnused Original Features (Count: 21): ['214875_x_at', '202283_at', '204499_at', '238073_at', '225112_at', '222271_at', '228498_at', '235612_at', '211404_s_at', '203818_s_at', '208248_x_at', '227139_s_at', '207643_s_at', '223499_at', '228569_at', '210260_s_at', '208704_x_at', '228662_at', '209263_x_at', '227689_at', '228988_at']\n",
            "\t\tThese features were not used to generate any of the output features. Add a feature generator compatible with these features to utilize them.\n",
            "\t\tFeatures can also be unused if they carry very little information, such as being categorical but having almost entirely unique values or being duplicates of other features.\n",
            "\t\tThese features do not need to be present at inference time.\n",
            "\t\t('object', []) : 21 | ['214875_x_at', '202283_at', '204499_at', '238073_at', '225112_at', ...]\n",
            "\tTypes of features in original data (raw dtype, special dtypes):\n",
            "\t\t('object', []) : 7 | ['241359_at', '204511_at', '1557275_a_at', '205364_at', '226535_at', ...]\n",
            "\tTypes of features in processed data (raw dtype, special dtypes):\n",
            "\t\t('category', []) : 7 | ['241359_at', '204511_at', '1557275_a_at', '205364_at', '226535_at', ...]\n",
            "\t0.3s = Fit runtime\n",
            "\t7 features in original data used to generate 7 features in processed data.\n",
            "\tTrain Data (Processed) Memory Usage: 0.0 MB (0.0% of available memory)\n",
            "Data preprocessing and feature engineering runtime = 0.42s ...\n",
            "AutoGluon will gauge predictive performance using evaluation metric: 'accuracy'\n",
            "\tTo change this, specify the eval_metric argument of fit()\n",
            "Fitting 11 L1 models ...\n",
            "Fitting model: NeuralNetFastAI_BAG_L1 ...\n",
            "No improvement since epoch 5: early stopping\n",
            "No improvement since epoch 1: early stopping\n",
            "\t0.525\t = Validation score   (accuracy)\n",
            "\t4.41s\t = Training   runtime\n",
            "\t0.09s\t = Validation runtime\n",
            "Fitting model: LightGBMXT_BAG_L1 ...\n",
            "/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:239: UserWarning: 'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.\n",
            "  _log_warning(\"'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. \"\n",
            "/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:239: UserWarning: 'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.\n",
            "  _log_warning(\"'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. \"\n",
            "/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:239: UserWarning: 'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.\n",
            "  _log_warning(\"'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. \"\n",
            "/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:239: UserWarning: 'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.\n",
            "  _log_warning(\"'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. \"\n",
            "/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:239: UserWarning: 'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.\n",
            "  _log_warning(\"'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. \"\n",
            "\t0.375\t = Validation score   (accuracy)\n",
            "\t1.41s\t = Training   runtime\n",
            "\t0.06s\t = Validation runtime\n",
            "Fitting model: LightGBM_BAG_L1 ...\n",
            "/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:239: UserWarning: 'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.\n",
            "  _log_warning(\"'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. \"\n",
            "/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:239: UserWarning: 'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.\n",
            "  _log_warning(\"'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. \"\n",
            "/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:239: UserWarning: 'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.\n",
            "  _log_warning(\"'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. \"\n",
            "/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:239: UserWarning: 'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.\n",
            "  _log_warning(\"'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. \"\n",
            "/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:239: UserWarning: 'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.\n",
            "  _log_warning(\"'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. \"\n",
            "\t0.375\t = Validation score   (accuracy)\n",
            "\t0.9s\t = Training   runtime\n",
            "\t0.06s\t = Validation runtime\n",
            "Fitting model: RandomForestGini_BAG_L1 ...\n",
            "\t0.55\t = Validation score   (accuracy)\n",
            "\t0.92s\t = Training   runtime\n",
            "\t0.11s\t = Validation runtime\n",
            "Fitting model: RandomForestEntr_BAG_L1 ...\n",
            "\t0.55\t = Validation score   (accuracy)\n",
            "\t0.87s\t = Training   runtime\n",
            "\t0.12s\t = Validation runtime\n",
            "Fitting model: CatBoost_BAG_L1 ...\n",
            "\t0.55\t = Validation score   (accuracy)\n",
            "\t2.94s\t = Training   runtime\n",
            "\t0.05s\t = Validation runtime\n",
            "Fitting model: ExtraTreesGini_BAG_L1 ...\n",
            "\t0.575\t = Validation score   (accuracy)\n",
            "\t0.76s\t = Training   runtime\n",
            "\t0.11s\t = Validation runtime\n",
            "Fitting model: ExtraTreesEntr_BAG_L1 ...\n",
            "\t0.575\t = Validation score   (accuracy)\n",
            "\t0.77s\t = Training   runtime\n",
            "\t0.1s\t = Validation runtime\n",
            "Fitting model: XGBoost_BAG_L1 ...\n",
            "\t0.525\t = Validation score   (accuracy)\n",
            "\t0.56s\t = Training   runtime\n",
            "\t0.04s\t = Validation runtime\n",
            "Fitting model: NeuralNetMXNet_BAG_L1 ...\n",
            "\t0.5\t = Validation score   (accuracy)\n",
            "\t5.28s\t = Training   runtime\n",
            "\t0.75s\t = Validation runtime\n",
            "Fitting model: LightGBMLarge_BAG_L1 ...\n",
            "/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:239: UserWarning: 'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.\n",
            "  _log_warning(\"'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. \"\n",
            "/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:239: UserWarning: 'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.\n",
            "  _log_warning(\"'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. \"\n",
            "/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:239: UserWarning: 'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.\n",
            "  _log_warning(\"'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. \"\n",
            "/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:239: UserWarning: 'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.\n",
            "  _log_warning(\"'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. \"\n",
            "/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:239: UserWarning: 'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.\n",
            "  _log_warning(\"'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. \"\n",
            "\t0.425\t = Validation score   (accuracy)\n",
            "\t0.92s\t = Training   runtime\n",
            "\t0.05s\t = Validation runtime\n",
            "Fitting model: WeightedEnsemble_L2 ...\n",
            "\t0.575\t = Validation score   (accuracy)\n",
            "\t0.29s\t = Training   runtime\n",
            "\t0.0s\t = Validation runtime\n",
            "AutoGluon training complete, total runtime = 22.47s ...\n",
            "Fitting 1 L1 models ...\n",
            "Fitting model: NeuralNetFastAI_BAG_L1_FULL ...\n",
            "\tStopping at the best epoch learned earlier - 10.\n",
            "\t0.3s\t = Training   runtime\n",
            "Fitting 1 L1 models ...\n",
            "Fitting model: LightGBMXT_BAG_L1_FULL ...\n",
            "/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:239: UserWarning: 'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.\n",
            "  _log_warning(\"'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. \"\n",
            "\t0.17s\t = Training   runtime\n",
            "Fitting 1 L1 models ...\n",
            "Fitting model: LightGBM_BAG_L1_FULL ...\n",
            "/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:239: UserWarning: 'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.\n",
            "  _log_warning(\"'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. \"\n",
            "\t0.17s\t = Training   runtime\n",
            "Fitting 1 L1 models ...\n",
            "Fitting model: RandomForestGini_BAG_L1_FULL ...\n",
            "\t0.55\t = Validation score   (accuracy)\n",
            "\t0.86s\t = Training   runtime\n",
            "\t0.1s\t = Validation runtime\n",
            "Fitting 1 L1 models ...\n",
            "Fitting model: RandomForestEntr_BAG_L1_FULL ...\n",
            "\t0.55\t = Validation score   (accuracy)\n",
            "\t0.86s\t = Training   runtime\n",
            "\t0.1s\t = Validation runtime\n",
            "Fitting 1 L1 models ...\n",
            "Fitting model: CatBoost_BAG_L1_FULL ...\n",
            "\t0.05s\t = Training   runtime\n",
            "Fitting 1 L1 models ...\n",
            "Fitting model: ExtraTreesGini_BAG_L1_FULL ...\n",
            "\t0.575\t = Validation score   (accuracy)\n",
            "\t0.87s\t = Training   runtime\n",
            "\t0.1s\t = Validation runtime\n",
            "Fitting 1 L1 models ...\n",
            "Fitting model: ExtraTreesEntr_BAG_L1_FULL ...\n",
            "\t0.575\t = Validation score   (accuracy)\n",
            "\t0.76s\t = Training   runtime\n",
            "\t0.1s\t = Validation runtime\n",
            "Fitting 1 L1 models ...\n",
            "Fitting model: XGBoost_BAG_L1_FULL ...\n",
            "\t0.04s\t = Training   runtime\n",
            "Fitting 1 L1 models ...\n",
            "Fitting model: NeuralNetMXNet_BAG_L1_FULL ...\n",
            "\t0.42s\t = Training   runtime\n",
            "Fitting 1 L1 models ...\n",
            "Fitting model: LightGBMLarge_BAG_L1_FULL ...\n",
            "/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:239: UserWarning: 'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.\n",
            "  _log_warning(\"'verbose_eval' argument is deprecated and will be removed in a future release of LightGBM. \"\n",
            "\t0.18s\t = Training   runtime\n",
            "Fitting model: WeightedEnsemble_L2_FULL ...\n",
            "\t0.575\t = Validation score   (accuracy)\n",
            "\t0.0s\t = Training   runtime\n",
            "\t0.0s\t = Validation runtime\n",
            "TabularPredictor saved. To load, use: predictor = TabularPredictor.load(\"good_quality_atrt/\")\n"
          ]
        }
      ],
      "source": [
        "#===========================================================================\n",
        "# Your code here\n",
        "# Train a classification model using AutoGluon TabularPrediction module with\n",
        "# features selected by the RandomForestClassifier\n",
        "#===========================================================================\n",
        "\n",
        "from autogluon.tabular import TabularPredictor\n",
        "\n",
        "train_data = x_train.loc[:, selected_genes].copy()\n",
        "train_data[\"Class\"] = y_train\n",
        "\n",
        "predictor = TabularPredictor(label=\"Class\", path=\"good_quality_atrt\").fit( \n",
        "    train_data=train_data,\n",
        "    presets=\"good_quality_faster_inference_only_refit\",\n",
        ")\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XhIzH-gWDSbe"
      },
      "source": [
        "Performance scores are calculated to evaluate the model. "
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "mnzalB9XBV9C"
      },
      "outputs": [],
      "source": [
        "from sklearn.metrics import (accuracy_score, balanced_accuracy_score, \n",
        "                             roc_auc_score, f1_score)\n",
        "def performance_scores(y_true, y_pred_score, y_pred=None):\n",
        "    # We can find which class has the highest score as its predicted class\n",
        "    if y_pred is None:\n",
        "        y_pred = y_pred_score.idxmax(axis=1)\n",
        "        \n",
        "    return {\n",
        "        \"accuracy\": accuracy_score(y_true, y_pred),\n",
        "        \"balanced_accuracy\": balanced_accuracy_score(y_true, y_pred),\n",
        "        #\"auroc\": roc_auc_score(y_true, y_pred_score[:, 1], average=\"weighted\",\n",
        "        #                       multi_class=\"ovr\"),\n",
        "        \"f1\": f1_score(y_true, y_pred, average=\"weighted\")\n",
        "    }"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "lB7FlxQoDbcD",
        "outputId": "28a71a92-3a45-4eb2-884b-6e8216d7e4f2"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "{'accuracy': 0.3333333333333333,\n",
              " 'balanced_accuracy': 0.3333333333333333,\n",
              " 'f1': 0.1818181818181818}"
            ]
          },
          "execution_count": null,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "\n",
        "test_data = x_test.loc[:, selected_genes].copy()\n",
        "y_pred_score = predictor.predict_proba(test_data).values\n",
        "y_pred = predictor.predict(test_data)\n",
        "y_pred\n",
        "performance_scores(y_test, y_pred_score, y_pred)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "5-edBhZ0dkvT",
        "outputId": "d88132a7-32bf-4889-f6ef-b7bccdc2f9c6"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "array([[0.02039674, 0.95528865, 0.02431464],\n",
              "       [0.52641743, 0.11188054, 0.361702  ],\n",
              "       [0.52641743, 0.11188054, 0.361702  ],\n",
              "       [0.52641743, 0.11188054, 0.361702  ],\n",
              "       [0.52641743, 0.11188054, 0.361702  ],\n",
              "       [0.52641743, 0.11188054, 0.361702  ],\n",
              "       [0.52641743, 0.11188054, 0.361702  ],\n",
              "       [0.52641743, 0.11188054, 0.361702  ],\n",
              "       [0.52641743, 0.11188054, 0.361702  ]], dtype=float32)"
            ]
          },
          "execution_count": null,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "y_pred_score"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "okONyxrpJvFU"
      },
      "source": [
        "Now also plot the confusion matrix to show correct and incorrect predictions in the test set"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "eIJyCZxSOgVC",
        "outputId": "85484002-577b-404c-b231-4f331bc6a48a"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "<sklearn.metrics._plot.confusion_matrix.ConfusionMatrixDisplay at 0x7f92ba35a110>"
            ]
          },
          "execution_count": null,
          "metadata": {},
          "output_type": "execute_result"
        },
        {
          "data": {
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAUQAAAEKCAYAAABquCzaAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAAdPElEQVR4nO3de7xVdZ3/8df7XAQUURNSRBQ1JgVGifip6E/C/GlSjTippTE29qucUsdudrH8eXv8Zsaspl+mjp3Un9fxbg0q460idUoLEIkDlpBXJAUhjYsi53zmj7U2bE/nnL2P7L3WOvu8nz3Wo70u+7s/Z5Ufv5f1/S5FBGZmBk15B2BmVhROiGZmKSdEM7OUE6KZWcoJ0cws5YRoZpZyQjSzfkfSYEm/lvSEpHZJF3RzzSBJt0haKukxSWMqleuEaGb90RvA+yPiAGAicLSkg7tc8ylgTUS8C/ge8K1KhTohmlm/E4m16W5runWdZTIDuDb9fDtwhCT1Vm5LTaPMiVqGhLbZPu8wCus9++2RdwjWAObPn7cqIkZsTRnNw/aM2LShqmtjw8p24PWyQ20R0VbakdQMzAPeBVwWEY91KWIU8DxARGyS9CqwM7Cqp99sjIS4zfYMevdH8w6jsP7rsUvzDsEawJBWPbu1ZcSmDVX/s/r6gstej4jJPZYV0QFMlLQj8GNJEyJi0dbE5yazmWVIoKbqtipFxJ+AnwNHdzm1HBgNIKkF2AF4pbeynBDNLDsCmpqr23orRhqR1gyRNAQ4Eniyy2WzgL9PPx8P/CwqrGbTEE1mM+tHeh/XqNZI4Nq0H7EJuDUi7pZ0ITA3ImYBVwHXS1oKrAZOrFSoE6KZZUh9ag73JCIWAu/p5vi5ZZ9fB07oS7lOiGaWrdrUEOvCCdHMsiNqUkOsFydEM8uQXEM0M9uswghynpwQzSxDtRlUqRcnRDPLjnCT2cxsM9cQzczATWYzsxIBzR5UMTNLuA/RzAzcZDYzK+caoplZyjVEMzOS2qFriGZmKU/dMzMDD6qYmZVzk9nMDK+HaGa2hZvMZmZbeFDFzCzlPkQzM9LnEN1kNjNLuIZoZpaQE6KZWekNAk6IZmYgoSYnxAFh0DYt3NP2BQa1ttDc0sysnz7ORW2z8w6rcB785WLO/u7tdHR2cvKMQ/jiKUflHVKhNPr9KXINsa7DPZJ2lrQg3f4oabmkJyQtkjSj7LoTJN2bfu5Ir18k6S5JO9Yzxlp6Y+MmZnzuEg6beRFTP/4vHDFlHJMnjMk7rELp6OjkKxffym3fP41Hbz2HO+6fx5N/WJF3WIUxEO6PpKq2CmWMlvRzSYsltUv6fDfXTJP0alkOOrdSbHVNiBHxSkRMjIiJwBXA9yLiAOBE4CJJgyUNBf4ZOD392ob0OxOA1WXH+4V1GzYC0NrSTGtLMxGRc0TFMq/9GfYePZwxuw9nm9YWPnLkJGb/YmHeYRXGQLg/tUiIwCbgyxExDjgYOF3SuG6ue7iUgyLiwkqF5tJkjohFku4CvgZsB1wXEcu6ufRXwP6ZBreVmprEnOu/xl67j+Cq2x5iXvuzeYdUKCtWvsqoXXbavL/bLjsxb9Ez+QVUMA1/f5RuWykiVgAr0s9/lrQEGAUs3ppy8+xDvACYD2wEJnc9KakZOAK4KuO4tkpnZzB15kUMGzqEG779GfbbZyRLljVWk8fs7RJV1f76VqY0BngP8Fg3p6dIegJ4ETgrItp7Kyu3hBgR6yTdAqyNiDfKTg2RtIAk2y8BHuju+5JOBU4FoHVonaPtu9fWbuDheb/niCnjnBDLjByxA8tfWrN5/8WX1jByxA45RlQsA+H+NDVV3VM3XNLcsv22iGgrvyDtcrsD+EJEvNbl+/OBPSNiraQPAj8BxvYaW7WR1UlnupXbkPY57klSue62DzEi2iJickRMVsuQOodZnZ13HMqwoUksgwe1cviB+/LUMy/lHFWxTBq3J8ueW8mzy1ex8c1N3PnAfKZP7Ve9InU1EO5PH/oQV5X+GU+3rsmwlSQZ3hgRd3b9nYh4LSLWpp9nA62ShvcWW2Efu4mI9ZLOBH4i6fKI2JR3TJXsOnwYl59/Ms1NTTQ1iR8/OJ/7HlmUd1iF0tLSzMVf/SjHnXkZHR3BzGMOZr99RuYdVmE0/P2pUR+ikox5FbAkIv61h2t2BV6KiJB0IEkF8JXeyi1sQgSIiMclLQROAq7PO55K2pe+yPv+7lt5h1F4Rx06nqMOHZ93GIXV6PenRn2IhwInA79Nu9gAvgHsARARVwDHA5+TtAnYAJwYFR77yCwhRsT5VR4b2mX/b+oXlZllqVaDKhHxCBXqmhFxKXBpX8otdA3RzBqPp+6ZmUHyBoECT91zQjSzTDkhmpmlnBDNzKjPTJVackI0s2wVNx86IZpZhtSnqXuZc0I0s0y5yWxmVlLcfOiEaGbZcg3RzAyqXQ07N06IZpYpJ0Qzs5TnMpuZpVxDNDMDL+5gZlYioMD50AnRzLLkUWYzs82aPKhiZkbah5h3ED1zQjSzzAjXEM3MNnMN0cws5UEVMzNwH6KZWYmQF4g1MytxDdHMLOU+RDMzcB+imVlJMpe5uBmxuL2bZtaQpOq23svQaEk/l7RYUrukz3dzjSRdImmppIWSJlWKzTVEM8tUjWaqbAK+HBHzJW0PzJP0QEQsLrtmOjA23Q4C/i39755jq0VkZmZV0Zb3qlTaehMRKyJifvr5z8ASYFSXy2YA10XiUWBHSSN7K9c1RDPLTB/XQxwuaW7ZfltEtP1FmdIY4D3AY11OjQKeL9t/IT22oqcfdEI0swz1aT3EVRExudfSpKHAHcAXIuK1rY3OCdHMMlWrQWZJrSTJ8MaIuLObS5YDo8v2d0+P9ch9iGaWHSWDKtVsvRaTVDOvApZExL/2cNks4BPpaPPBwKsR0WNzGVxDNLMM1fA5xEOBk4HfSlqQHvsGsAdARFwBzAY+CCwF1gOfrFSoE6KZZaoWCTEiHiHJr71dE8DpfSnXCdHMMlXgiSpOiGaWrSJP3XNCNLPseHEHM7NEskBscTOiE6KZZaqpwFVEJ0Qzy1SB86ETopllR/KgipnZZgXuQuw5IUr6ARA9nY+IM+sSkZk1tP46qDK3l3NmZn0mkpHmouoxIUbEteX7kraNiPX1D8nMGlmBK4iVV7uRNEXSYuDJdP8ASZfXPTIzazxVrpad18BLNct//T/gA8ArABHxBDC1nkGZWeOqxUum6qWqUeaIeL5Lxu6oTzhm1shE/38w+3lJhwCRrlD7eZIXupiZ9VmRR5mraTJ/lmRNsVHAi8BE+rjGmJkZVN9cLmyTOSJWATMziMXMBoAiN5mrGWXeW9JdklZKelnSf0jaO4vgzKzxqMotD9U0mf8duBUYCewG3AbcVM+gzKxx9ffHbraNiOsjYlO63QAMrndgZtZ4klHm6rY89DaX+R3px/+U9HXgZpK5zR8jeZuVmVnfqP8uEDuPJAGWov+HsnMBnF2voMyscfXL5b8iYq8sAzGzxldqMhdVVTNVJE0AxlHWdxgR19UrKDNrXP2yhlgi6TxgGklCnA1MBx4BnBDNrM+Kmw6rG2U+HjgC+GNEfBI4ANihrlGZWUOSoLlJVW15qKbJvCEiOiVtkjQMeBkYXee4+qVB27RwT9sXGNTaQnNLM7N++jgXtXlAvqsHf7mYs797Ox2dnZw84xC+eMpReYdUKI1+f4rcZK6mhjhX0o7Aj0hGnucDv6qmcEnflNQuaaGkBZIOkjRH0uSya8ZIWpR+nibp7i5lXCPp+Kr/ohy9sXETMz53CYfNvIipH/8XjpgyjskTxuQdVqF0dHTylYtv5bbvn8ajt57DHffP48k/rMg7rMIYCPenVnOZJV2dzp5b1MP5aZJeTXPPAknnViqzmrnMp6Ufr5B0LzAsIhZWEewU4MPApIh4Q9JwYJtK3+vv1m3YCEBrSzOtLc1E9PhamgFpXvsz7D16OGN2Hw7AR46cxOxfLGTfvUfmHFkxNPr9EarlXOZrgEvpfTzj4Yj4cLUF9vZg9qTezkXE/ApljwRWRcQbsHmRiEJXl2uhqUnMuf5r7LX7CK667SHmtT+bd0iFsmLlq4zaZafN+7vtshPzFj2TX0AF0/D3p4Yr2UTEQ5LG1Ka0RG81xO/2Fgvw/gpl3w+cK+n3wIPALRHxi/TcjZI2pJ+3ATrLvneYpAVl+3sAb2lGA0g6FTgVgNahFULJTmdnMHXmRQwbOoQbvv0Z9ttnJEuWNVaTx2xr9KFSNFxS+cvu2iKirY8/N0XSEyRLF54VEe29Xdzbg9mH9/GHu35/raT3AocBhwO3pFMAAWZGxFxI+hB5a8J7SxVX0jU9lN8GtAE0bfvOwrVLX1u7gYfn/Z4jpoxzQiwzcsQOLH9pzeb9F19aw8gRfmihpNHvj4Dm6hPiqoiYXPmyHs0H9kxz0QeBnwBje/tCNYMqb1tEdETEnIg4DzgDOK6ev5e3nXccyrChQwAYPKiVww/cl6eeeSnnqIpl0rg9WfbcSp5dvoqNb27izgfmM33q/nmHVRgD4f5ktbhDRLwWEWvTz7OB1nQso0dVzVR5OyS9G+iMiKfSQxOBZ4EJ9frNvO06fBiXn38yzU1NNDWJHz84n/se6XYAbMBqaWnm4q9+lOPOvIyOjmDmMQez3z6NMWBQCwPh/mT1iKGkXYGXIiIkHUhSAXylt+/ULSECQ4EfpI/sbAKWkvT53V7H38xV+9IXed/ffSvvMArvqEPHc9Sh4/MOo7Aa+f4kj9TUJiNKuolkFt1wSS8A5wGtABFxBcmkks9J2gRsAE6MCo99VDN1TySvENg7Ii6UtAewa0T8urfvRcQ84JBuTk3rct0zpLXGiJgDzOly/pRKMZpZ/1GrGmJEnFTh/KUkj+VUrZo+xMuBKUDpx/8MXNaXHzEzK+nXL5kCDoqISZIeB4iINZIa/gFrM6s9AS0Ffha5moT4pqRmkmcPkTSCtz43aGZWtQLnw6oS4iXAj4F3Svonko7Kc+oalZk1JKmmU/dqrpq5zDdKmkeyBJiAYyNiSd0jM7OGVOB8WNUo8x7AeuCu8mMR8Vw9AzOzxtTfXyFwD1teNjUY2Av4HdCYD0qZWd0Iclv8tRrVNJn/unw/XQXntB4uNzPrWY7vXK5Gn2eqRMR8SQfVIxgza3wq8FtVqulD/FLZbhMwiWQpHTOzPmmE15BuX/Z5E0mf4h31CcfMGl2/TYjpA9nbR8RZGcVjZg2uyKvm9/YKgZaI2CTp0CwDMrPGlbyGNO8oetZbDfHXJP2FCyTNAm4D1pVORsSddY7NzBpQv56pQvLs4Ssk71ApPY8YgBOimfVJfx5UeWc6wryILYmwpHDvMDGz/qHAFcReE2IzyarX3YXvhGhmb4No6qfPIa6IiAszi8TMGp7ovzXEAodtZv2SoKXAnYi9JcQjMovCzAaEfltDjIjVWQZiZgNDf3/sxsysZgqcD50QzSw7orpXfebFCdHMsiM3mc3MgNJMFSdEMzOg2M/zOSGaWaYKXEEsdP+mmTUcIVW3VSxJulrSy5IW9XBeki6RtFTSwvR9UL1yQjSzzJRGmavZqnANcHQv56cDY9PtVODfKhXohGhmmWqSqtoqiYiHgN4mkMwArovEo8COkkb2VmZD9CFu+46dmPCxE/IOo7DuXuR3glXy4Qm75R3CwKA+vUJguKS5ZfttEdHWh18bBTxftv9CemxFT19oiIRoZv1DHx/MXhURk+sWTDecEM0sUxm+ZGo5MLpsf/f0WI/ch2hmmVKVWw3MAj6RjjYfDLwaET02l8E1RDPLkIDmGtUQJd0ETCPpa3wBOA9oBYiIK4DZwAeBpcB64JOVynRCNLNM1arFHBEnVTgfwOl9KdMJ0cwyJFTgyXtOiGaWqSJP3XNCNLPMJI/dFDcjOiGaWXbkGqKZ2WZeD9HMjNICsXlH0TMnRDPLlEeZzcxSBW4xOyGaWbZcQzQzw32IZmZbVLn4a16cEM0sU8VNh06IZpYhv5fZzKxMcdOhE6KZZa3AGdEJ0cwy5SazmVmquOnQCdHMslbgjOiEaGaZSV4gVdyM6IRoZtnxeohmZlsUOB86IZpZlpTli+r7zAnRzDJV4HzohGhm2RFuMpuZbVHgjOiEaGaZ8mM3A8SI7Qfxjen7stN2rUTA3QtXcMf85XmHVSg/uupuHn9iKcOGbctF//fUvMMppAd/uZizv3s7HZ2dnDzjEL54ylF5h1RTRe5DbKpXwZJC0g1l+y2SVkq6W9J4Sb+XNKTs/D2STpI0VNIPJS2TNE/SHEkH1SvOWuroDC6fs4xT/v9cTrvxcY6duBt77rxt3mEVymH/c3+++qUT8w6jsDo6OvnKxbdy2/dP49Fbz+GO++fx5B9W5B1W7aTPIVazVSxKOlrS7yQtlfT1bs6fkuacBen26Upl1i0hAuuACWVJ70hgOUBEtAN3At8EkHQs0BoRNwFXAquBsRHxXuCTwPA6xlkzq9dt5KmX1wKw4c0Onl29nuFDB+UcVbHs++492G7o4LzDKKx57c+w9+jhjNl9ONu0tvCRIycx+xcL8w6rplTlf3otQ2oGLgOmA+OAkySN6+bSWyJiYrpdWSm2eiZEgNnAh9LPJwE3lZ27EDhB0kTgIuB0SfsABwHnREQnQEQ8HRH31DnOmtt12CDGvnMoS1a8lnco1o+sWPkqo3bZafP+brvsxIqVr+YYUW2JmtUQDwSWRsQfImIjcDMwY2vjq3dCvBk4UdJgYH/gsdKJiFgPnAU8BNwcEU8B44EFEdFR57jqakhrExccM55Lf76M9Rv79Z9iVnOqcqtgFPB82f4L6bGujpO0UNLtkkZXKrSuCTEiFgJjSGqHs7s5fxfwJ+DyvpYt6VRJcyXNfXNdcf4N2twkLjhmPA8ueZmHn1qVdzjWz4wcsQPLX1qzef/Fl9YwcsQOOUZUB9VnxOGlf8bTra+jcHcBYyJif+AB4NpKX6h3DRFgFvAd3tpcLteZbgDtwAFp/0CvIqItIiZHxOTW7Yrzf5ivfuCveG71em6b90LeoVg/NGncnix7biXPLl/Fxjc3cecD85k+df+8w6qppvTNe5U2YFXpn/F0aysrZjlQXuPbPT22WUS8EhFvpLtXAu+tFFsWj91cDfwpIn4raVpvF0bEMklzgQsk/Z+ICEljgPH9oR/xr0cN4wPjd2XZyrVc+Ynk3v/o4ad57OnVOUdWHJdd8ROWPPksa9du4Mwv/YCPHHsY06ZOzDuswmhpaebir36U4868jI6OYOYxB7PfPiPzDqumavTUzW+AsZL2IkmEJwIff8vvSCMjojREfwywpFKhdU+IEfECcEkfvvJp4LvAUkkbgFXAV+oRW639dvlrTPvOL/IOo9BO/+yxeYdQeEcdOp6jDh2fdxj1U4OMGBGbJJ0B3Ac0A1dHRLukC4G5ETELOFPSMcAmkidXTqlUbt0SYkQM7ebYHGBOl2Njuuy/BnymXnGZWX5quUBsRMymy9hERJxb9vls4Oy+lOmZKmaWHS8Qa2a2RYHzoROimWXJC8SamW1W4HzohGhm2fECsWZm5QqcEZ0QzSxTXiDWzCzlPkQzMwBBkxOimVlJcTOiE6KZZaa0QGxROSGaWaYKnA+dEM0sW64hmpmlPHXPzCxV3HTohGhmGar2nct5cUI0s0x5poqZWUlx86ETopllq8D50AnRzLK0+RWjheSEaGaZKfpMlSxeVG9m1i+4hmhmmSpyDdEJ0cwy5cduzMzA72U2Mysp+qCKE6KZZcpNZjOzVJFriH7sxswypSq3iuVIR0v6naSlkr7ezflBkm5Jzz8maUylMp0QzSxbNciIkpqBy4DpwDjgJEnjulz2KWBNRLwL+B7wrUqhOSGaWWYENElVbRUcCCyNiD9ExEbgZmBGl2tmANemn28HjlCF1WkVEX3/qwpG0krg2bzj6GI4sCrvIArM96eyot2jPSNixNYUIOlekr+rGoOB18v22yKiLS3neODoiPh0un8ycFBEnFH2W4vSa15I95el1/R4TxtiUGVr/0eqB0lzI2Jy3nEUle9PZY14jyLi6Lxj6I2bzGbWHy0HRpft754e6/YaSS3ADsArvRXqhGhm/dFvgLGS9pK0DXAiMKvLNbOAv08/Hw/8LCr0ETZEk7mg2vIOoOB8fyrzPepBRGySdAZwH9AMXB0R7ZIuBOZGxCzgKuB6SUuB1SRJs1cNMahiZlYLbjKbmaWcEM3MUk6IfSRpZ0kL0u2PkpZLekLSIkkzyq47IX3mCkkd6fWLJN0lacf8/oL6k/RNSe2SFqZ/90GS5kiaXHbNmPQ5MSRNk3R3lzKuSZ81aziSQtINZfstklZKulvSeEm/lzSk7Pw9kk6SNFTSDyUtkzQvvacH5fNXNCYPqvRRRLwCTASQdD6wNiK+I2kCcJuk+0ju6z8DpWeuNkRE6TvXAqcD/5R17FmQNAX4MDApIt6QNBzYJuewimYdMEHSkIjYABxJ+shIOjBwJ/BN4BxJxwKtEXGTpJuBp4GxEdEpaS+SaWtWI06INRIRiyTdBXwN2A64LiKWdXPpr4D9Mw0uWyOBVRHxBkBpVkCFGVMD0WzgQyRTyk4CbgIOS89dCDwu6XbgIuBvJO0DHATMjIhOgIh4miRBWo24yVxbFwAfJ5lwfnHXk+mE9CP4y+elGsn9wOi02Xe5pPeVnbux1N1AkhDKHVbWFbEAOCaziPNxM3CipMEk/4J8rHQiItYDZwEPATdHxFPAeGBBRHTkEexA4RpiDUXEOkm3kDSj3yg7NST9h3wUsAR4IJcAMxARayW9l6S2czhwS9nSTDMjYi4kfYhAeb/hwxHx4dKOpGsyCTgnEbEwvQcn8Zf/ciAi7pL0J+DyjEMb0FxDrL3OdCtX6kPck2TBj9MzjypDEdEREXMi4jzgDOC4vGMqqFnAd0iay90p//9SO3BA2sqwOnFCzFDaFDoT+HI6t7LhSHq3pLFlhyZSvJWIiuJq4IKI+G2lC9P+6LnABaUlrNKR+g/VOcYBxQkxYxHxOLCQpKnUiIYC10paLGkhySjo+fmGVEwR8UJEXNKHr3wa2AVYmj6ydA3wcj1iG6g8dc/MLOUaoplZygnRzCzlhGhmlnJCNDNLOSGamaWcEAeILivu3CZp260oa/NKNJKu7OZ9uOXXTpN0yNv4jWfShSGqOt7lmrV9/K3zJZ3V1xit8TghDhwbImJiREwANgKfLT/5dh8Uj4hPR8TiXi6ZBvQ5IZrlwQlxYHoYeFdae3tY0ixgsaRmSd+W9Jt0LcN/AFDiUkm/k/Qg8M5SQeXrHEo6WtL8dH3In6ZzdT8LfDGtnR4maYSkO9Lf+I2kQ9Pv7izp/nQdxStJpjj2StJP0nUB2yWd2uXc99LjP5U0Ij22j6R70+88LGnfWtxMaxwNOX3MepbWBKcD96aHJgETIuLpNKm8GhH/Q9Ig4L8k3Q+8B3g3yayTXYDFJNPOyssdAfwImJqW9Y6IWC3pCtI1I9Pr/h34XkQ8ImkPkpcE7QecBzwSERem09E+VcWf87/T3xgC/EbSHel6lduRvGjoi5LOTcs+g+SlTZ+NiKfShVUvB97/Nm6jNSgnxIGjtOIOJDXEq0iasr9O19UDOArYX1tWqt4BGAtMBW5Kl556UdLPuin/YOChUlkRsbqHOP4XMK5sfcRhkoamv/GR9Lv3SFpTxd90pqS/TT+PTmN9hWRBhFvS4zcAd6a/cQjJIr6l7w+q4jdsAHFCHDg2r9pdkiaGdeWHgH+MiPu6XPfBGsbRBBwcEa93E0vVJE0jSa5TImK9pDnA4B4uj/R3/9T1HpiVcx+ilbsP+JykVgBJfyVpO5KFSj+W9jGOJFnnsKtHgalKlrVH0jvS438Gti+77n7gH0s7kkoJ6iGSxXWRNB3YqUKsOwBr0mS4L0kNtaSJ5MXkpGU+EhGvAU9LOiH9DUk6oMJv2ADjhGjlriTpH5yfrqbyQ5JWxI+Bp9Jz15G8BuEtImIlcCpJ8/QJtjRZ7wL+tjSoQrL82eR00GYxW0a7LyBJqO0kTefnKsR6L9AiaQnJMvuPlp1bBxyY/g3vJ1mSH2Am8Kk0vnZgBmZlvNqNmVnKNUQzs5QToplZygnRzCzlhGhmlnJCNDNLOSGamaWcEM3MUv8NrYaJpuriQL8AAAAASUVORK5CYII=\n",
            "text/plain": [
              "<Figure size 432x288 with 2 Axes>"
            ]
          },
          "metadata": {},
          "output_type": "display_data"
        }
      ],
      "source": [
        "#============================================================================\n",
        "# Your code here\n",
        "# Plot confusion matrix for the trained model\n",
        "#============================================================================\n",
        "\n",
        "from sklearn.metrics import confusion_matrix, ConfusionMatrixDisplay\n",
        "\n",
        "cm = confusion_matrix(y_test, y_pred)\n",
        "\n",
        "disp = ConfusionMatrixDisplay(cm, display_labels=[\"TYR\", \"SHH\",\"MYC\"])\n",
        "disp.plot(cmap=\"Blues\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "efSKXkmTEZpb"
      },
      "source": [
        "The models are displayed here in a leader board to show which performed best.\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "VXGsi5c5LuSE",
        "outputId": "20e90f35-47f2-4d27-c371-719c8be70c68"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "                           model  score_val  pred_time_val  fit_time  pred_time_val_marginal  fit_time_marginal  stack_level  can_infer  fit_order\n",
            "0     ExtraTreesEntr_BAG_L1_FULL      0.575       0.100577  0.760374                0.100577           0.760374            1       True         20\n",
            "1     ExtraTreesGini_BAG_L1_FULL      0.575       0.101893  0.867148                0.101893           0.867148            1       True         19\n",
            "2          ExtraTreesEntr_BAG_L1      0.575       0.102436  0.772450                0.102436           0.772450            1      False          8\n",
            "3          ExtraTreesGini_BAG_L1      0.575       0.114720  0.764169                0.114720           0.764169            1      False          7\n",
            "4            WeightedEnsemble_L2      0.575       0.115330  1.050670                0.000610           0.286501            2      False         12\n",
            "5                CatBoost_BAG_L1      0.550       0.050498  2.939611                0.050498           2.939611            1      False          6\n",
            "6   RandomForestEntr_BAG_L1_FULL      0.550       0.101501  0.863880                0.101501           0.863880            1       True         17\n",
            "7   RandomForestGini_BAG_L1_FULL      0.550       0.102737  0.861766                0.102737           0.861766            1       True         16\n",
            "8        RandomForestGini_BAG_L1      0.550       0.113915  0.923613                0.113915           0.923613            1      False          4\n",
            "9        RandomForestEntr_BAG_L1      0.550       0.117148  0.871294                0.117148           0.871294            1      False          5\n",
            "10                XGBoost_BAG_L1      0.525       0.039397  0.562150                0.039397           0.562150            1      False          9\n",
            "11        NeuralNetFastAI_BAG_L1      0.525       0.090173  4.406495                0.090173           4.406495            1      False          1\n",
            "12         NeuralNetMXNet_BAG_L1      0.500       0.749770  5.283608                0.749770           5.283608            1      False         10\n",
            "13          LightGBMLarge_BAG_L1      0.425       0.046106  0.923977                0.046106           0.923977            1      False         11\n",
            "14             LightGBMXT_BAG_L1      0.375       0.055240  1.409494                0.055240           1.409494            1      False          2\n",
            "15               LightGBM_BAG_L1      0.375       0.055256  0.897764                0.055256           0.897764            1      False          3\n",
            "16      WeightedEnsemble_L2_FULL        NaN       0.103026  0.867606                0.001133           0.000458            2       True         24\n",
            "17           XGBoost_BAG_L1_FULL        NaN            NaN  0.035558                     NaN           0.035558            1       True         21\n",
            "18    NeuralNetMXNet_BAG_L1_FULL        NaN            NaN  0.424809                     NaN           0.424809            1       True         22\n",
            "19   NeuralNetFastAI_BAG_L1_FULL        NaN            NaN  0.298980                     NaN           0.298980            1       True         13\n",
            "20          LightGBM_BAG_L1_FULL        NaN            NaN  0.172760                     NaN           0.172760            1       True         15\n",
            "21        LightGBMXT_BAG_L1_FULL        NaN            NaN  0.165250                     NaN           0.165250            1       True         14\n",
            "22     LightGBMLarge_BAG_L1_FULL        NaN            NaN  0.183092                     NaN           0.183092            1       True         23\n",
            "23          CatBoost_BAG_L1_FULL        NaN            NaN  0.051059                     NaN           0.051059            1       True         18\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>model</th>\n",
              "      <th>score_val</th>\n",
              "      <th>pred_time_val</th>\n",
              "      <th>fit_time</th>\n",
              "      <th>pred_time_val_marginal</th>\n",
              "      <th>fit_time_marginal</th>\n",
              "      <th>stack_level</th>\n",
              "      <th>can_infer</th>\n",
              "      <th>fit_order</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>ExtraTreesEntr_BAG_L1_FULL</td>\n",
              "      <td>0.575</td>\n",
              "      <td>0.100577</td>\n",
              "      <td>0.760374</td>\n",
              "      <td>0.100577</td>\n",
              "      <td>0.760374</td>\n",
              "      <td>1</td>\n",
              "      <td>True</td>\n",
              "      <td>20</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>ExtraTreesGini_BAG_L1_FULL</td>\n",
              "      <td>0.575</td>\n",
              "      <td>0.101893</td>\n",
              "      <td>0.867148</td>\n",
              "      <td>0.101893</td>\n",
              "      <td>0.867148</td>\n",
              "      <td>1</td>\n",
              "      <td>True</td>\n",
              "      <td>19</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>ExtraTreesEntr_BAG_L1</td>\n",
              "      <td>0.575</td>\n",
              "      <td>0.102436</td>\n",
              "      <td>0.772450</td>\n",
              "      <td>0.102436</td>\n",
              "      <td>0.772450</td>\n",
              "      <td>1</td>\n",
              "      <td>False</td>\n",
              "      <td>8</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>ExtraTreesGini_BAG_L1</td>\n",
              "      <td>0.575</td>\n",
              "      <td>0.114720</td>\n",
              "      <td>0.764169</td>\n",
              "      <td>0.114720</td>\n",
              "      <td>0.764169</td>\n",
              "      <td>1</td>\n",
              "      <td>False</td>\n",
              "      <td>7</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>WeightedEnsemble_L2</td>\n",
              "      <td>0.575</td>\n",
              "      <td>0.115330</td>\n",
              "      <td>1.050670</td>\n",
              "      <td>0.000610</td>\n",
              "      <td>0.286501</td>\n",
              "      <td>2</td>\n",
              "      <td>False</td>\n",
              "      <td>12</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>CatBoost_BAG_L1</td>\n",
              "      <td>0.550</td>\n",
              "      <td>0.050498</td>\n",
              "      <td>2.939611</td>\n",
              "      <td>0.050498</td>\n",
              "      <td>2.939611</td>\n",
              "      <td>1</td>\n",
              "      <td>False</td>\n",
              "      <td>6</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>RandomForestEntr_BAG_L1_FULL</td>\n",
              "      <td>0.550</td>\n",
              "      <td>0.101501</td>\n",
              "      <td>0.863880</td>\n",
              "      <td>0.101501</td>\n",
              "      <td>0.863880</td>\n",
              "      <td>1</td>\n",
              "      <td>True</td>\n",
              "      <td>17</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>RandomForestGini_BAG_L1_FULL</td>\n",
              "      <td>0.550</td>\n",
              "      <td>0.102737</td>\n",
              "      <td>0.861766</td>\n",
              "      <td>0.102737</td>\n",
              "      <td>0.861766</td>\n",
              "      <td>1</td>\n",
              "      <td>True</td>\n",
              "      <td>16</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>RandomForestGini_BAG_L1</td>\n",
              "      <td>0.550</td>\n",
              "      <td>0.113915</td>\n",
              "      <td>0.923613</td>\n",
              "      <td>0.113915</td>\n",
              "      <td>0.923613</td>\n",
              "      <td>1</td>\n",
              "      <td>False</td>\n",
              "      <td>4</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>RandomForestEntr_BAG_L1</td>\n",
              "      <td>0.550</td>\n",
              "      <td>0.117148</td>\n",
              "      <td>0.871294</td>\n",
              "      <td>0.117148</td>\n",
              "      <td>0.871294</td>\n",
              "      <td>1</td>\n",
              "      <td>False</td>\n",
              "      <td>5</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>XGBoost_BAG_L1</td>\n",
              "      <td>0.525</td>\n",
              "      <td>0.039397</td>\n",
              "      <td>0.562150</td>\n",
              "      <td>0.039397</td>\n",
              "      <td>0.562150</td>\n",
              "      <td>1</td>\n",
              "      <td>False</td>\n",
              "      <td>9</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>NeuralNetFastAI_BAG_L1</td>\n",
              "      <td>0.525</td>\n",
              "      <td>0.090173</td>\n",
              "      <td>4.406495</td>\n",
              "      <td>0.090173</td>\n",
              "      <td>4.406495</td>\n",
              "      <td>1</td>\n",
              "      <td>False</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>NeuralNetMXNet_BAG_L1</td>\n",
              "      <td>0.500</td>\n",
              "      <td>0.749770</td>\n",
              "      <td>5.283608</td>\n",
              "      <td>0.749770</td>\n",
              "      <td>5.283608</td>\n",
              "      <td>1</td>\n",
              "      <td>False</td>\n",
              "      <td>10</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>LightGBMLarge_BAG_L1</td>\n",
              "      <td>0.425</td>\n",
              "      <td>0.046106</td>\n",
              "      <td>0.923977</td>\n",
              "      <td>0.046106</td>\n",
              "      <td>0.923977</td>\n",
              "      <td>1</td>\n",
              "      <td>False</td>\n",
              "      <td>11</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>LightGBMXT_BAG_L1</td>\n",
              "      <td>0.375</td>\n",
              "      <td>0.055240</td>\n",
              "      <td>1.409494</td>\n",
              "      <td>0.055240</td>\n",
              "      <td>1.409494</td>\n",
              "      <td>1</td>\n",
              "      <td>False</td>\n",
              "      <td>2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>LightGBM_BAG_L1</td>\n",
              "      <td>0.375</td>\n",
              "      <td>0.055256</td>\n",
              "      <td>0.897764</td>\n",
              "      <td>0.055256</td>\n",
              "      <td>0.897764</td>\n",
              "      <td>1</td>\n",
              "      <td>False</td>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>16</th>\n",
              "      <td>WeightedEnsemble_L2_FULL</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.103026</td>\n",
              "      <td>0.867606</td>\n",
              "      <td>0.001133</td>\n",
              "      <td>0.000458</td>\n",
              "      <td>2</td>\n",
              "      <td>True</td>\n",
              "      <td>24</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17</th>\n",
              "      <td>XGBoost_BAG_L1_FULL</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.035558</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.035558</td>\n",
              "      <td>1</td>\n",
              "      <td>True</td>\n",
              "      <td>21</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>18</th>\n",
              "      <td>NeuralNetMXNet_BAG_L1_FULL</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.424809</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.424809</td>\n",
              "      <td>1</td>\n",
              "      <td>True</td>\n",
              "      <td>22</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>19</th>\n",
              "      <td>NeuralNetFastAI_BAG_L1_FULL</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.298980</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.298980</td>\n",
              "      <td>1</td>\n",
              "      <td>True</td>\n",
              "      <td>13</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>20</th>\n",
              "      <td>LightGBM_BAG_L1_FULL</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.172760</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.172760</td>\n",
              "      <td>1</td>\n",
              "      <td>True</td>\n",
              "      <td>15</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>21</th>\n",
              "      <td>LightGBMXT_BAG_L1_FULL</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.165250</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.165250</td>\n",
              "      <td>1</td>\n",
              "      <td>True</td>\n",
              "      <td>14</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22</th>\n",
              "      <td>LightGBMLarge_BAG_L1_FULL</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.183092</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.183092</td>\n",
              "      <td>1</td>\n",
              "      <td>True</td>\n",
              "      <td>23</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23</th>\n",
              "      <td>CatBoost_BAG_L1_FULL</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.051059</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.051059</td>\n",
              "      <td>1</td>\n",
              "      <td>True</td>\n",
              "      <td>18</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "                           model  score_val  ...  can_infer  fit_order\n",
              "0     ExtraTreesEntr_BAG_L1_FULL      0.575  ...       True         20\n",
              "1     ExtraTreesGini_BAG_L1_FULL      0.575  ...       True         19\n",
              "2          ExtraTreesEntr_BAG_L1      0.575  ...      False          8\n",
              "3          ExtraTreesGini_BAG_L1      0.575  ...      False          7\n",
              "4            WeightedEnsemble_L2      0.575  ...      False         12\n",
              "5                CatBoost_BAG_L1      0.550  ...      False          6\n",
              "6   RandomForestEntr_BAG_L1_FULL      0.550  ...       True         17\n",
              "7   RandomForestGini_BAG_L1_FULL      0.550  ...       True         16\n",
              "8        RandomForestGini_BAG_L1      0.550  ...      False          4\n",
              "9        RandomForestEntr_BAG_L1      0.550  ...      False          5\n",
              "10                XGBoost_BAG_L1      0.525  ...      False          9\n",
              "11        NeuralNetFastAI_BAG_L1      0.525  ...      False          1\n",
              "12         NeuralNetMXNet_BAG_L1      0.500  ...      False         10\n",
              "13          LightGBMLarge_BAG_L1      0.425  ...      False         11\n",
              "14             LightGBMXT_BAG_L1      0.375  ...      False          2\n",
              "15               LightGBM_BAG_L1      0.375  ...      False          3\n",
              "16      WeightedEnsemble_L2_FULL        NaN  ...       True         24\n",
              "17           XGBoost_BAG_L1_FULL        NaN  ...       True         21\n",
              "18    NeuralNetMXNet_BAG_L1_FULL        NaN  ...       True         22\n",
              "19   NeuralNetFastAI_BAG_L1_FULL        NaN  ...       True         13\n",
              "20          LightGBM_BAG_L1_FULL        NaN  ...       True         15\n",
              "21        LightGBMXT_BAG_L1_FULL        NaN  ...       True         14\n",
              "22     LightGBMLarge_BAG_L1_FULL        NaN  ...       True         23\n",
              "23          CatBoost_BAG_L1_FULL        NaN  ...       True         18\n",
              "\n",
              "[24 rows x 9 columns]"
            ]
          },
          "execution_count": null,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "predictor.leaderboard()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "Zy72mcqr9EWW",
        "outputId": "243a3ad1-0f61-40e0-b5d6-b1849ef4c1e5"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "Computing feature importance via permutation shuffling for 28 features using 40 rows with 10 shuffle sets...\n",
            "\t43.76s\t= Expected runtime (4.38s per shuffle set)\n",
            "\t2.68s\t= Actual runtime (Completed 10 of 10 shuffle sets)\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>importance</th>\n",
              "      <th>stddev</th>\n",
              "      <th>p_value</th>\n",
              "      <th>n</th>\n",
              "      <th>p99_high</th>\n",
              "      <th>p99_low</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>1557275_a_at</th>\n",
              "      <td>0.1000</td>\n",
              "      <td>0.023570</td>\n",
              "      <td>1.480476e-07</td>\n",
              "      <td>10</td>\n",
              "      <td>0.124223</td>\n",
              "      <td>0.075777</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>226535_at</th>\n",
              "      <td>0.1650</td>\n",
              "      <td>0.042817</td>\n",
              "      <td>3.376672e-07</td>\n",
              "      <td>10</td>\n",
              "      <td>0.209003</td>\n",
              "      <td>0.120997</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>241359_at</th>\n",
              "      <td>0.0875</td>\n",
              "      <td>0.033850</td>\n",
              "      <td>9.313427e-06</td>\n",
              "      <td>10</td>\n",
              "      <td>0.122287</td>\n",
              "      <td>0.052713</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>225877_at</th>\n",
              "      <td>0.0400</td>\n",
              "      <td>0.047434</td>\n",
              "      <td>1.288140e-02</td>\n",
              "      <td>10</td>\n",
              "      <td>0.088748</td>\n",
              "      <td>-0.008748</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "              importance    stddev       p_value   n  p99_high   p99_low\n",
              "1557275_a_at      0.1000  0.023570  1.480476e-07  10  0.124223  0.075777\n",
              "226535_at         0.1650  0.042817  3.376672e-07  10  0.209003  0.120997\n",
              "241359_at         0.0875  0.033850  9.313427e-06  10  0.122287  0.052713\n",
              "225877_at         0.0400  0.047434  1.288140e-02  10  0.088748 -0.008748"
            ]
          },
          "execution_count": null,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "featimp = predictor.feature_importance(train_data, num_shuffle_sets=10)\n",
        "featimp[featimp.importance > 0].sort_values(\"p_value\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "6hUxjCt4ZfdG"
      },
      "source": [
        "The best model in this dataset was the 'ExtraTreesEntr_BAG_L1_FULL' and we can see that the most important features were all the at genes, which is expected in atrt patients. "
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "A9RSEX_7L71P"
      },
      "source": [
        "Additional features from the meta file can be added to the analysis, but it can be seen that they are not selected and make no difference on the AutoML model. "
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "iiI303PWLz6v"
      },
      "outputs": [],
      "source": [
        "feature_columns = [\n",
        "    'Age', 'Gender'\n",
        "]\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "D0-aHU9CM7Vh"
      },
      "outputs": [],
      "source": [
        "# Creating training and testing data. \n",
        "\n",
        "x_train = pd.concat([\n",
        "    meta.loc[meta.subset == \"train\", feature_columns + [\"SampleGeo\"]].set_index(\"SampleGeo\"),\n",
        "    ge[meta[meta.subset == \"train\"].SampleGeo].transpose()\n",
        "], axis=1)\n",
        "y_train = target[(meta.subset == \"train\")].values\n",
        "x_test = pd.concat([\n",
        "    meta.loc[meta.subset == \"test\", feature_columns + [\"SampleGeo\"]].set_index(\"SampleGeo\"),\n",
        "    ge[meta[meta.subset == \"test\"].SampleGeo].transpose()\n",
        "], axis=1)\n",
        "y_test = target[(meta.subset == \"test\")].values"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "-vYhQk9UPZfD"
      },
      "outputs": [],
      "source": [
        "# for categorical variable, we simply encode it as 0 for male, 1 for female\n",
        "x_train[\"Gender\"] = (x_train.Gender == \"F\").astype(int)\n",
        "x_test[\"Gender\"] = (x_test.Gender == \"F\").astype(int)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "tnel6L_oO1uZ"
      },
      "outputs": [],
      "source": [
        "# for missing data, we'll use the mean imputation\n",
        "from sklearn.impute import SimpleImputer\n",
        "\n",
        "imputer = SimpleImputer()\n",
        "x_train_mi = imputer.fit_transform(x_train)\n",
        "x_test_mi = imputer.transform(x_test)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "qBkQTSnCUzOR"
      },
      "outputs": [],
      "source": [
        "midf = pd.DataFrame({\n",
        "    \"feature\": x_train.columns,\n",
        "    \"mi\": mutual_info_classif(x_train_mi, y_train)\n",
        "}).sort_values(\"mi\", ascending=False)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "fGdb6MlXbFh3"
      },
      "outputs": [],
      "source": [
        "plt.plot(np.arange(100), midf.mi.iloc[:100])"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "PyiesG7HQGMg"
      },
      "outputs": [],
      "source": [
        "selected_features = midf.loc[midf.mi > 0.62, \"feature\"]"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "6afIossYQQms"
      },
      "outputs": [],
      "source": [
        "# there are no features in the metadata selected based on MI\n",
        "[x for x in meta.columns if x in selected_features]"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "CcSCOzsRNLPV"
      },
      "outputs": [],
      "source": [
        "\n",
        "from autogluon.tabular import TabularPredictor\n",
        "\n",
        "train_data = x_train[selected_features].copy()\n",
        "train_data[\"Class\"] = y_train\n",
        "\n",
        "predictor = TabularPredictor(\n",
        "    label=\"Class\", path=\"good_quality_atrt_meta\"\n",
        ").fit( \n",
        "    train_data=train_data,\n",
        "    presets=\"good_quality_faster_inference_only_refit\",\n",
        ")\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "Ln6zLMm7NiE9"
      },
      "outputs": [],
      "source": [
        "test_data = x_test[selected_features].copy()\n",
        "y_pred_score = predictor.predict_proba(test_data).values\n",
        "y_pred = predictor.predict(test_data)\n",
        "\n",
        "performance_scores(y_test, y_pred_score, y_pred)\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "NELjrieONmM6"
      },
      "outputs": [],
      "source": [
        "from sklearn.metrics import confusion_matrix, ConfusionMatrixDisplay\n",
        "cm = confusion_matrix(y_test, y_pred)\n",
        "\n",
        "disp = ConfusionMatrixDisplay(cm, display_labels=[\"TYR\", \"SHH\",\"MYC\"])\n",
        "disp.plot(cmap=\"Blues\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ASvUk-jAbeTD"
      },
      "source": [
        "The features in the metadata did not have high enough MI with the severity and thus none of the features were selected into model training. The performance of the model remains the same given the same feature selection criteria."
      ]
    }
  ],
  "metadata": {
    "colab": {
      "collapsed_sections": [],
      "name": "Fall2021GroupProjectDDNR.ipynb",
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.6.6"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}